---
title: How To Get Genomewide Position-Specific Scores
author: ZERO
date: '2018-04-27'
slug: how-to-get-genomewide-position-specific-scores
categories:
  - bioinformatics
tags:
  - phast
thumbnailImagePosition: left
thumbnailImage: /post/2018-04-27-how-to-get-genomewide-position-specific-scores_files/download.jpg
metaAlignment: center
coverMeta: out
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(
	echo = TRUE,#show code and output
	message = FALSE,
	warning = FALSE,
	engine.path="/Users/zero/anaconda3/bin/python"
)
```

## load packages
```{r}
#source("http://bioconductor.org/biocLite.R")
#biocLite("GenomicScores")
library(GenomicScores)
```


## Retrieval of genomic scores through annotation packages
There are currently four different annotation packages that store genomic scores and can be accessed using the _[GenomicScores](http://bioconductor.org/packages/GenomicScores)_ package


<strong style="color: darkred;">Annotation packages </strong> 	<strong style="color: darkred;">Description</strong> 
1. _[phastCons100way.UCSC.hg19](http://bioconductor.org/packages/phastCons100way.UCSC.hg19)_	phastCons scores derived from the alignment of the human genome (hg19) to other 99 vertebrate species.
2. _[phastCons100way.UCSC.hg38](http://bioconductor.org/packages/phastCons100way.UCSC.hg38)_	phastCons scores derived from the alignment of the human genome (hg38) to other 99 vertebrate species.
3. _[phastCons7way.UCSC.hg38](http://bioconductor.org/packages/phastCons7way.UCSC.hg38)_  phastCons scores derived from the alignment of the human genome (hg38) to other 6 mammal species.
4. _[fitCons.UCSC.hg19](http://bioconductor.org/packages/fitCons.UCSC.hg19)_	fitCons scores: fitness consequences of functional annotation for the human genome (hg19).

### <strong style="color: darkred;">To retrieve genomic scores for specific positions we should use the function scores()</strong> 
```{r}
library(phastCons100way.UCSC.hg19)
gsco <- phastCons100way.UCSC.hg19
class(gsco)
scores(gsco, GRanges(seqnames="chr22",
                     IRanges(start=50967020:50967021, width=1)))
gsco
#citation(gsco) now cann't run
provider(gsco)
providerVersion(gsco)
organism(gsco)
seqlevelsStyle(gsco)
```

## Retrieval of genomic scores through <strong style="color: darkred;">AnnotationHub resources</strong> 

Another way to retrieve genomic scores is by using the _[AnnotationHub](http://bioconductor.org/packages/AnnotationHub)_, which is a web resource that provides a central location where genomic files (e.g., VCF, bed, wig) and other resources from standard (e.g., UCSC, Ensembl) and distributed sites, can be found. A Bioconductor _[AnnotationHub](http://bioconductor.org/packages/AnnotationHub)_ web resource creates and manages a local cache of files retrieved by the user, helping with quick and reproducible access.
```{r eval=FALSE, include=FALSE}
availableGScores()
```

```{r out.width="30%", fig.align = "center", fig.margin=TRUE, echo=FALSE, eval=TRUE}
knitr::include_graphics("/post/2018-04-27-how-to-get-genomewide-position-specific-scores_files/Screen Shot 2018-04-27 at 15.35.21.png")
```

```{r eval=FALSE, include=TRUE}
gsco <- getGScores("phastCons7way.UCSC.hg38")
scores(gsco, GRanges(seqnames="chr22",IRanges(start=20967020:20967025, width=1)))
```


## Retrieval of <strong style="color: darkred;">multiple scores</strong>  per genomic position

## Summarization of genomic scores
```{r}
library(dplyr)
gsco <- phastCons100way.UCSC.hg19
gr <- GRanges(seqnames="chr22", IRanges(start=50967020:50967025, width=1))

#mean
scores(gsco, gr) %>% 
  .$scores %>% 
  mean()
#median
scores(gsco, gr) %>% 
  .$scores %>% 
  median()
#min
scores(gsco, gr) %>% 
  .$scores %>% 
  min()

scores(gsco, gr) %>% 
  .$scores %>% 
  hist()
```

## <strong style="color: darkred;">Annotating variants with genomic scores</strong> 

A typical use case of the _[GenomicScores](http://bioconductor.org/packages/GenomicScores)_ package is in the context of annotating variants with genomic scores, such as phastCons conservation scores. For this purpose, we load the _[VariantAnnotaiton](http://bioconductor.org/packages/VariantAnnotaiton)_ and _[TxDb.Hsapiens.UCSC.hg19.knownGene](http://bioconductor.org/packages/TxDb.Hsapiens.UCSC.hg19.knownGene)_ packages. The former will allow us to read a VCF file and annotate it, and the latter contains the gene annotations from UCSC that will be used in this process.

Let’s load one of the sample VCF files that form part of the _[VariantAnnotation](http://bioconductor.org/packages/VariantAnnotation)_ package.
```{r}
library(VariantAnnotation)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
fl <- system.file("extdata", "chr22.vcf.gz", package="VariantAnnotation")
vcf <- readVcf(fl, "hg19")
seqlevelsStyle(vcf)
```

```{r}
txdb <- TxDb.Hsapiens.UCSC.hg19.knownGene
seqlevelsStyle(txdb)
```

Because the chromosome nomenclature from the VCF file (NCBI) is different from the one with the gene annotations (UCSC) we use the <strong style="color: darkred;">seqlevelsStyle()</strong>  function to force our variants having the chromosome nomenclature of the gene annotations.
```{r}
seqlevelsStyle(vcf) <- seqlevelsStyle(txdb)
```

We <strong style="color: darkred;">annotate the location of variants</strong>  using the function <strong style="color: darkred;">locateVariants()</strong>  from the VariantAnnotation package.

```{r}
loc <- locateVariants(vcf, txdb, AllVariants())
loc[1:3]
table(loc$LOCATION)
```

## <strong style="color: darkred;">Annotate phastCons conservation scores</strong> 

on the variants and store those annotations as an additional metadata column of the GRanges object. For this specific purpose we should the argument scores.only=TRUE that makes the scores() method to return the genomic scores as a numeric vector instead as a metadata column in the input ranges object.

```{r}
loc$PHASTCONS <- scores(gsco, loc, scores.only=TRUE)
loc[1:3]
```

Using the following code we can examine <strong style="color: darkred;">the distribution of phastCons conservation scores</strong>  of variants across the different annotated regions
```{r}
x <- split(loc$PHASTCONS, loc$LOCATION)
mask <- elementNROWS(x) > 0
boxplot(x[mask], ylab="phastCons score", las=1, cex.axis=1.2, cex.lab=1.5, col="gray")
points(1:length(x[mask])+0.25, 
       sapply(x[mask], mean, na.rm=TRUE),
       pch=23, bg="red")
```


##**Links**
- [An introduction to the GenomicScores package](http://bioconductor.org/packages/release/bioc/vignettes/GenomicScores/inst/doc/GenomicScores.html)


