

  
    
  


  





  

<!DOCTYPE html>
<html lang="en-us">
  <head>
    
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="generator" content="Hugo 0.32.2 with theme Tranquilpeak 0.4.1-BETA">
    <title>How To Get Genomewide Position-Specific Scores</title>
    <meta name="author" content="Jixing Liu">
    <meta name="keywords" content=", R">

    <link rel="icon" href="img/R_py.png">
    

    
    <meta name="description" content="load packages #source(&quot;http://bioconductor.org/biocLite.R&quot;) #biocLite(&quot;GenomicScores&quot;) library(GenomicScores)  Retrieval of genomic scores through annotation packages There are currently four different annotation packages that store genomic scores and can be accessed using theÂ GenomicScoresÂ package
Annotation packages  Description 1. phastCons100way.UCSC.hg19 phastCons scores derived from the alignment of the human genome (hg19) to other 99 vertebrate species. 2. phastCons100way.UCSC.hg38 phastCons scores derived from the alignment of the human genome (hg38) to other 99 vertebrate species.">
    <meta property="og:description" content="load packages #source(&quot;http://bioconductor.org/biocLite.R&quot;) #biocLite(&quot;GenomicScores&quot;) library(GenomicScores)  Retrieval of genomic scores through annotation packages There are currently four different annotation packages that store genomic scores and can be accessed using theÂ GenomicScoresÂ package
Annotation packages  Description 1. phastCons100way.UCSC.hg19 phastCons scores derived from the alignment of the human genome (hg19) to other 99 vertebrate species. 2. phastCons100way.UCSC.hg38 phastCons scores derived from the alignment of the human genome (hg38) to other 99 vertebrate species.">
    <meta property="og:type" content="blog">
    <meta property="og:title" content="How To Get Genomewide Position-Specific Scores">
    <meta property="og:url" content="/2018/04/how-to-get-genomewide-position-specific-scores/">
    <meta property="og:site_name" content="student zero ">
    <meta name="twitter:card" content="summary">
    <meta name="twitter:title" content="student zero ">
    <meta name="twitter:description" content="load packages #source(&quot;http://bioconductor.org/biocLite.R&quot;) #biocLite(&quot;GenomicScores&quot;) library(GenomicScores)  Retrieval of genomic scores through annotation packages There are currently four different annotation packages that store genomic scores and can be accessed using theÂ GenomicScoresÂ package
Annotation packages  Description 1. phastCons100way.UCSC.hg19 phastCons scores derived from the alignment of the human genome (hg19) to other 99 vertebrate species. 2. phastCons100way.UCSC.hg38 phastCons scores derived from the alignment of the human genome (hg38) to other 99 vertebrate species.">
    
      <meta name="twitter:creator" content="@studentZero475">
    
    

    
    

    
      <meta property="og:image" content="https://i.loli.net/2018/02/26/5a937bb148b02.jpg">
    

    
      <meta property="og:image" content="/post/2018-04-27-how-to-get-genomewide-position-specific-scores_files/download.jpg">
    
    
    

    

    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.4/jquery.fancybox.min.css" integrity="sha256-vuXZ9LGmmwtjqFX1F+EKin1ThZMub58gKULUyf0qECk=" crossorigin="anonymous" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.4/helpers/jquery.fancybox-thumbs.min.css" integrity="sha256-SEa4XYAHihTcEP1f5gARTB2K26Uk8PsndQYHQC1f4jU=" crossorigin="anonymous" />
    
    
    <link rel="stylesheet" href="/css/style-hbs5om8csx9a8yrv5hnhpi2qqdv4ykuslajwbramhqxvleqbfklxgek50hye.min.css" />
    
    

    
      
    
    
  </head>

  <body>
    <div id="blog">
      <link rel="stylesheet" href="/css/monokai-sublime.css" rel="stylesheet" id="theme-stylesheet">
<script src="/js/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>

<header id="header" data-behavior="4">
  <i id="btn-open-sidebar" class="fa fa-lg fa-bars"></i>
  <div class="header-title">
    <a class="header-title-link" href="/">student zero </a>
  </div>
  
    
      <a class="header-right-picture "
         href="/#about">
    
    
    
      
        <img class="header-picture" src="https://i.loli.net/2018/02/26/5a937bb148b02.jpg" alt="Author&#39;s picture" />
      
    
    </a>
  
</header>

      <nav id="sidebar" data-behavior="4">
  <div class="sidebar-container">
    
      <div class="sidebar-profile">
        <h1 class="sidebar-profile-title">How To Get Genomewide Position-Specific Scores</h1>
        <a href="/#about">
          <img class="sidebar-profile-picture" src="https://i.loli.net/2018/02/26/5a937bb148b02.jpg" alt="Author&#39;s picture" />
        </a>
        <h4 class="sidebar-profile-name">Jixing Liu</h4>
        
          <h5 class="sidebar-profile-bio">Reading And Writing</h5>
        
      </div>
    
    <ul class="sidebar-buttons">
      
  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="/">
    
      <i class="sidebar-button-icon fa fa-lg fa-home"></i>
      
      <span class="sidebar-button-desc">Home</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="/archives/">
    
      <i class="sidebar-button-icon fa fa-lg fa-archive"></i>
      
      <span class="sidebar-button-desc">Archives</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="/categories/">
    
      <i class="sidebar-button-icon fa fa-lg fa-bookmark"></i>
      
      <span class="sidebar-button-desc">Categories</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="/tags/">
    
      <i class="sidebar-button-icon fa fa-lg fa-tags"></i>
      
      <span class="sidebar-button-desc">Tags</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="/#about">
    
      <i class="sidebar-button-icon fa fa-lg fa-question"></i>
      
      <span class="sidebar-button-desc">About</span>
    </a>
  </li>


    </ul>
    <ul class="sidebar-buttons">
      
  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://twitter.com/studentZero475">
    
      <i class="sidebar-button-icon fa fa-lg fa-twitter"></i>
      
      <span class="sidebar-button-desc">Twitter</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://www.r-bloggers.com/">
    
      <i class="sidebar-button-icon fa fa-lg fa-book"></i>
      
      <span class="sidebar-button-desc">R-bloggers</span>
    </a>
  </li>


    </ul>
    <ul class="sidebar-buttons">
      

    </ul>
  </div>
</nav>

      

      <div id="main" data-behavior="4"
        class="
               hasCoverMetaOut
               ">
        <article class="post" itemscope itemType="http://schema.org/BlogPosting">
          
          
            <div class="post-header main-content-wrap text-center">
  
    <h1 class="post-title" itemprop="headline">
      How To Get Genomewide Position-Specific Scores
    </h1>
  
  
  <div class="postShorten-meta post-meta">
    
      <time itemprop="datePublished" datetime="2018-04-27T00:00:00Z">
        
  April 27, 2018

      </time>
    
    
  
  
    <span>in</span>
    
      <a class="category-link" href="/categories/bioinformatics">bioinformatics</a>
    
  


  </div>

</div>
          
          <div class="post-content markdown" itemprop="articleBody">
            <div class="main-content-wrap">
              <div id="load-packages" class="section level2">
<h2>load packages</h2>
<pre class="r"><code>#source(&quot;http://bioconductor.org/biocLite.R&quot;)
#biocLite(&quot;GenomicScores&quot;)
library(GenomicScores)</code></pre>
</div>
<div id="retrieval-of-genomic-scores-through-annotation-packages" class="section level2">
<h2>Retrieval of genomic scores through annotation packages</h2>
<p>There are currently four different annotation packages that store genomic scores and can be accessed using theÂ <em><a href="http://bioconductor.org/packages/GenomicScores">GenomicScores</a></em>Â package</p>
<p><strong style="color: darkred;">Annotation packages </strong> <strong style="color: darkred;">Description</strong> 1. <em><a href="http://bioconductor.org/packages/phastCons100way.UCSC.hg19">phastCons100way.UCSC.hg19</a></em> phastCons scores derived from the alignment of the human genome (hg19) to other 99 vertebrate species. 2. <em><a href="http://bioconductor.org/packages/phastCons100way.UCSC.hg38">phastCons100way.UCSC.hg38</a></em> phastCons scores derived from the alignment of the human genome (hg38) to other 99 vertebrate species. 3. <em><a href="http://bioconductor.org/packages/phastCons7way.UCSC.hg38">phastCons7way.UCSC.hg38</a></em> phastCons scores derived from the alignment of the human genome (hg38) to other 6 mammal species. 4. <em><a href="http://bioconductor.org/packages/fitCons.UCSC.hg19">fitCons.UCSC.hg19</a></em> fitCons scores: fitness consequences of functional annotation for the human genome (hg19).</p>
<div id="to-retrieve-genomic-scores-for-specific-positions-we-should-use-the-function-scores" class="section level3">
<h3><strong style="color: darkred;">To retrieve genomic scores for specific positions we should use the function scores()</strong></h3>
<pre class="r"><code>library(phastCons100way.UCSC.hg19)
gsco &lt;- phastCons100way.UCSC.hg19
class(gsco)</code></pre>
<pre><code>## [1] &quot;GScores&quot;
## attr(,&quot;package&quot;)
## [1] &quot;GenomicScores&quot;</code></pre>
<pre class="r"><code>scores(gsco, GRanges(seqnames=&quot;chr22&quot;,
                     IRanges(start=50967020:50967021, width=1)))</code></pre>
<pre><code>## GRanges object with 2 ranges and 1 metadata column:
##       seqnames               ranges strand |    scores
##          &lt;Rle&gt;            &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
##   [1]    chr22 [50967020, 50967020]      * |         1
##   [2]    chr22 [50967021, 50967021]      * |         1
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<pre class="r"><code>gsco</code></pre>
<pre><code>## GScores object 
## # organism: Homo sapiens (UCSC)
## # provider: UCSC
## # provider version: 09Feb2014
## # download date: Mar 17, 2017
## # loaded sequences: chr19_gl000208_random, chr22
## # maximum abs. error: 0.05</code></pre>
<pre class="r"><code>#citation(gsco) now cann&#39;t run
provider(gsco)</code></pre>
<pre><code>## [1] &quot;UCSC&quot;</code></pre>
<pre class="r"><code>providerVersion(gsco)</code></pre>
<pre><code>## [1] &quot;09Feb2014&quot;</code></pre>
<pre class="r"><code>organism(gsco)</code></pre>
<pre><code>## [1] &quot;Homo sapiens&quot;</code></pre>
<pre class="r"><code>seqlevelsStyle(gsco)</code></pre>
<pre><code>## [1] &quot;UCSC&quot;</code></pre>
</div>
</div>
<div id="retrieval-of-genomic-scores-through-annotationhub-resources" class="section level2">
<h2>Retrieval of genomic scores through <strong style="color: darkred;">AnnotationHub resources</strong></h2>
<p>Another way to retrieve genomic scores is by using theÂ <em><a href="http://bioconductor.org/packages/AnnotationHub">AnnotationHub</a></em>, which is a web resource that provides a central location where genomic files (e.g., VCF, bed, wig) and other resources from standard (e.g., UCSC, Ensembl) and distributed sites, can be found. A BioconductorÂ <em><a href="http://bioconductor.org/packages/AnnotationHub">AnnotationHub</a></em>Â web resource creates and manages a local cache of files retrieved by the user, helping with quick and reproducible access.</p>
<p><img src="/post/2018-04-27-how-to-get-genomewide-position-specific-scores_files/Screen Shot 2018-04-27 at 15.35.21.png" width="30%" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gsco &lt;- getGScores(&quot;phastCons7way.UCSC.hg38&quot;)
scores(gsco, GRanges(seqnames=&quot;chr22&quot;,IRanges(start=20967020:20967025, width=1)))</code></pre>
</div>
<div id="retrieval-of-multiple-scores-per-genomic-position" class="section level2">
<h2>Retrieval of <strong style="color: darkred;">multiple scores</strong> per genomic position</h2>
</div>
<div id="summarization-of-genomic-scores" class="section level2">
<h2>Summarization of genomic scores</h2>
<pre class="r"><code>library(dplyr)
gsco &lt;- phastCons100way.UCSC.hg19
gr &lt;- GRanges(seqnames=&quot;chr22&quot;, IRanges(start=50967020:50967025, width=1))

#mean
scores(gsco, gr) %&gt;% 
  .$scores %&gt;% 
  mean()</code></pre>
<pre><code>## [1] 0.8</code></pre>
<pre class="r"><code>#median
scores(gsco, gr) %&gt;% 
  .$scores %&gt;% 
  median()</code></pre>
<pre><code>## [1] 1</code></pre>
<pre class="r"><code>#min
scores(gsco, gr) %&gt;% 
  .$scores %&gt;% 
  min()</code></pre>
<pre><code>## [1] 0</code></pre>
<pre class="r"><code>scores(gsco, gr) %&gt;% 
  .$scores %&gt;% 
  hist()</code></pre>
<p><img src="/post/2018-04-27-how-to-get-genomewide-position-specific-scores_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
</div>
<div id="annotating-variants-with-genomic-scores" class="section level2">
<h2><strong style="color: darkred;">Annotating variants with genomic scores</strong></h2>
<p>A typical use case of theÂ <em><a href="http://bioconductor.org/packages/GenomicScores">GenomicScores</a></em>Â package is in the context of annotating variants with genomic scores, such as phastCons conservation scores. For this purpose, we load theÂ <em><a href="http://bioconductor.org/packages/VariantAnnotaiton">VariantAnnotaiton</a></em>Â andÂ <em><a href="http://bioconductor.org/packages/TxDb.Hsapiens.UCSC.hg19.knownGene">TxDb.Hsapiens.UCSC.hg19.knownGene</a></em>Â packages. The former will allow us to read a VCF file and annotate it, and the latter contains the gene annotations from UCSC that will be used in this process.</p>
<p>Letâ€™s load one of the sample VCF files that form part of theÂ <em><a href="http://bioconductor.org/packages/VariantAnnotation">VariantAnnotation</a></em>Â package.</p>
<pre class="r"><code>library(VariantAnnotation)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
fl &lt;- system.file(&quot;extdata&quot;, &quot;chr22.vcf.gz&quot;, package=&quot;VariantAnnotation&quot;)
vcf &lt;- readVcf(fl, &quot;hg19&quot;)
seqlevelsStyle(vcf)</code></pre>
<pre><code>## [1] &quot;NCBI&quot;    &quot;Ensembl&quot;</code></pre>
<pre class="r"><code>txdb &lt;- TxDb.Hsapiens.UCSC.hg19.knownGene
seqlevelsStyle(txdb)</code></pre>
<pre><code>## [1] &quot;UCSC&quot;</code></pre>
<p>Because the chromosome nomenclature from the VCF file (NCBI) is different from the one with the gene annotations (UCSC) we use the <strong style="color: darkred;">seqlevelsStyle()</strong> function to force our variants having the chromosome nomenclature of the gene annotations.</p>
<pre class="r"><code>seqlevelsStyle(vcf) &lt;- seqlevelsStyle(txdb)</code></pre>
<p>We <strong style="color: darkred;">annotate the location of variants</strong> using the function <strong style="color: darkred;">locateVariants()</strong> from the VariantAnnotation package.</p>
<pre class="r"><code>loc &lt;- locateVariants(vcf, txdb, AllVariants())
loc[1:3]</code></pre>
<pre><code>## GRanges object with 3 ranges and 9 metadata columns:
##    seqnames               ranges strand | LOCATION  LOCSTART    LOCEND
##       &lt;Rle&gt;            &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt; &lt;integer&gt; &lt;integer&gt;
##       chr22 [50300078, 50300078]      - |   intron     10763     10763
##       chr22 [50300086, 50300086]      - |   intron     10755     10755
##       chr22 [50300101, 50300101]      - |   intron     10740     10740
##      QUERYID        TXID         CDSID      GENEID       PRECEDEID
##    &lt;integer&gt; &lt;character&gt; &lt;IntegerList&gt; &lt;character&gt; &lt;CharacterList&gt;
##            1       75253                     79087                
##            2       75253                     79087                
##            3       75253                     79087                
##           FOLLOWID
##    &lt;CharacterList&gt;
##                   
##                   
##                   
##   -------
##   seqinfo: 1 sequence from hg19 genome; no seqlengths</code></pre>
<pre class="r"><code>table(loc$LOCATION)</code></pre>
<pre><code>## 
## spliceSite     intron    fiveUTR   threeUTR     coding intergenic 
##         11      22572        309       1368       2822       2867 
##   promoter 
##       2864</code></pre>
</div>
<div id="annotate-phastcons-conservation-scores" class="section level2">
<h2><strong style="color: darkred;">Annotate phastCons conservation scores</strong></h2>
<p>on the variants and store those annotations as an additional metadata column of the GRanges object. For this specific purpose we should the argument scores.only=TRUE that makes the scores() method to return the genomic scores as a numeric vector instead as a metadata column in the input ranges object.</p>
<pre class="r"><code>loc$PHASTCONS &lt;- scores(gsco, loc, scores.only=TRUE)
loc[1:3]</code></pre>
<pre><code>## GRanges object with 3 ranges and 10 metadata columns:
##    seqnames               ranges strand | LOCATION  LOCSTART    LOCEND
##       &lt;Rle&gt;            &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt; &lt;integer&gt; &lt;integer&gt;
##       chr22 [50300078, 50300078]      - |   intron     10763     10763
##       chr22 [50300086, 50300086]      - |   intron     10755     10755
##       chr22 [50300101, 50300101]      - |   intron     10740     10740
##      QUERYID        TXID         CDSID      GENEID       PRECEDEID
##    &lt;integer&gt; &lt;character&gt; &lt;IntegerList&gt; &lt;character&gt; &lt;CharacterList&gt;
##            1       75253                     79087                
##            2       75253                     79087                
##            3       75253                     79087                
##           FOLLOWID PHASTCONS
##    &lt;CharacterList&gt; &lt;numeric&gt;
##                          0.0
##                          0.1
##                          0.0
##   -------
##   seqinfo: 1 sequence from hg19 genome; no seqlengths</code></pre>
<p>Using the following code we can examine <strong style="color: darkred;">the distribution of phastCons conservation scores</strong> of variants across the different annotated regions</p>
<pre class="r"><code>x &lt;- split(loc$PHASTCONS, loc$LOCATION)
mask &lt;- elementNROWS(x) &gt; 0
boxplot(x[mask], ylab=&quot;phastCons score&quot;, las=1, cex.axis=1.2, cex.lab=1.5, col=&quot;gray&quot;)
points(1:length(x[mask])+0.25, 
       sapply(x[mask], mean, na.rm=TRUE),
       pch=23, bg=&quot;red&quot;)</code></pre>
<p><img src="/post/2018-04-27-how-to-get-genomewide-position-specific-scores_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
</div>
<div id="links" class="section level2">
<h2><strong>Links</strong></h2>
<ul>
<li><a href="http://bioconductor.org/packages/release/bioc/vignettes/GenomicScores/inst/doc/GenomicScores.html">An introduction to the GenomicScores package</a></li>
</ul>
</div>

              
            </div>
          </div>
          <div id="post-footer" class="post-footer main-content-wrap">
            
              
                
                
                  <div class="post-footer-tags">
                    <span class="text-color-light text-small">TAGGED IN</span><br/>
                    
  <a class="tag tag--primary tag--small" href="//tags/phast/">phast</a>

                  </div>
                
              
            
            
  <div class="post-actions-wrap">
      <nav >
        <ul class="post-actions post-action-nav">
            <li class="post-action">
              
                <a class="post-action-btn btn btn--default tooltip--top" href="/2018/05/r-vs-python-representing-data-in-r-python-equivalent/" data-tooltip="R vs python">
              
                <i class="fa fa-angle-left"></i>
                <span class="hide-xs hide-sm text-small icon-ml">NEXT</span>
              </a>
            </li>
            <li class="post-action">
              
                <a class="post-action-btn btn btn--default tooltip--top" href="/2018/04/purrr-adverb/" data-tooltip="purrr: adverb">
              
                <span class="hide-xs hide-sm text-small icon-mr">PREVIOUS</span>
                <i class="fa fa-angle-right"></i>
              </a>
            </li>
        </ul>
      </nav>
    <ul class="post-actions post-action-share" >
      
        <li class="post-action hide-lg hide-md hide-sm">
          <a class="post-action-btn btn btn--default btn-open-shareoptions" href="#btn-open-shareoptions">
            <i class="fa fa-share-alt"></i>
          </a>
        </li>
        
          <li class="post-action hide-xs">
            <a class="post-action-btn btn btn--default" target="new" href="https://www.facebook.com/sharer/sharer.php?u=/2018/04/how-to-get-genomewide-position-specific-scores/">
              <i class="fa fa-facebook-official"></i>
            </a>
          </li>
        
          <li class="post-action hide-xs">
            <a class="post-action-btn btn btn--default" target="new" href="https://twitter.com/intent/tweet?text=/2018/04/how-to-get-genomewide-position-specific-scores/">
              <i class="fa fa-twitter"></i>
            </a>
          </li>
        
          <li class="post-action hide-xs">
            <a class="post-action-btn btn btn--default" target="new" href="https://plus.google.com/share?url=/2018/04/how-to-get-genomewide-position-specific-scores/">
              <i class="fa fa-google-plus"></i>
            </a>
          </li>
        
      
      
        <li class="post-action">
          <a class="post-action-btn btn btn--default" href="#disqus_thread">
            <i class="fa fa-comment-o"></i>
          </a>
        </li>
      
      <li class="post-action">
        
          <a class="post-action-btn btn btn--default" href="#">
        
          <i class="fa fa-list"></i>
        </a>
      </li>
    </ul>
  </div>


            
              
                <div id="disqus_thread">
  <noscript>Please enable JavaScript to view the <a href="//disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
</div>
              
            
          </div>
        </article>
        <footer id="footer" class="main-content-wrap">
  <span class="copyrights">
    &copy; 2018 Jixing Liu. All Rights Reserved
  </span>
</footer>

      </div>
      <div id="bottom-bar" class="post-bottom-bar" data-behavior="4">
        
  <div class="post-actions-wrap">
      <nav >
        <ul class="post-actions post-action-nav">
            <li class="post-action">
              
                <a class="post-action-btn btn btn--default tooltip--top" href="/2018/05/r-vs-python-representing-data-in-r-python-equivalent/" data-tooltip="R vs python">
              
                <i class="fa fa-angle-left"></i>
                <span class="hide-xs hide-sm text-small icon-ml">NEXT</span>
              </a>
            </li>
            <li class="post-action">
              
                <a class="post-action-btn btn btn--default tooltip--top" href="/2018/04/purrr-adverb/" data-tooltip="purrr: adverb">
              
                <span class="hide-xs hide-sm text-small icon-mr">PREVIOUS</span>
                <i class="fa fa-angle-right"></i>
              </a>
            </li>
        </ul>
      </nav>
    <ul class="post-actions post-action-share" >
      
        <li class="post-action hide-lg hide-md hide-sm">
          <a class="post-action-btn btn btn--default btn-open-shareoptions" href="#btn-open-shareoptions">
            <i class="fa fa-share-alt"></i>
          </a>
        </li>
        
          <li class="post-action hide-xs">
            <a class="post-action-btn btn btn--default" target="new" href="https://www.facebook.com/sharer/sharer.php?u=/2018/04/how-to-get-genomewide-position-specific-scores/">
              <i class="fa fa-facebook-official"></i>
            </a>
          </li>
        
          <li class="post-action hide-xs">
            <a class="post-action-btn btn btn--default" target="new" href="https://twitter.com/intent/tweet?text=/2018/04/how-to-get-genomewide-position-specific-scores/">
              <i class="fa fa-twitter"></i>
            </a>
          </li>
        
          <li class="post-action hide-xs">
            <a class="post-action-btn btn btn--default" target="new" href="https://plus.google.com/share?url=/2018/04/how-to-get-genomewide-position-specific-scores/">
              <i class="fa fa-google-plus"></i>
            </a>
          </li>
        
      
      
        <li class="post-action">
          <a class="post-action-btn btn btn--default" href="#disqus_thread">
            <i class="fa fa-comment-o"></i>
          </a>
        </li>
      
      <li class="post-action">
        
          <a class="post-action-btn btn btn--default" href="#">
        
          <i class="fa fa-list"></i>
        </a>
      </li>
    </ul>
  </div>


      </div>
      <div id="share-options-bar" class="share-options-bar" data-behavior="4">
  <i id="btn-close-shareoptions" class="fa fa-close"></i>
  <ul class="share-options">
    
      <li class="share-option">
        <a class="share-option-btn" target="new" href="https://www.facebook.com/sharer/sharer.php?u=%2F2018%2F04%2Fhow-to-get-genomewide-position-specific-scores%2F">
          <i class="fa fa-facebook-official"></i><span>Share on Facebook</span>
        </a>
      </li>
    
      <li class="share-option">
        <a class="share-option-btn" target="new" href="https://twitter.com/intent/tweet?text=%2F2018%2F04%2Fhow-to-get-genomewide-position-specific-scores%2F">
          <i class="fa fa-twitter"></i><span>Share on Twitter</span>
        </a>
      </li>
    
      <li class="share-option">
        <a class="share-option-btn" target="new" href="https://plus.google.com/share?url=%2F2018%2F04%2Fhow-to-get-genomewide-position-specific-scores%2F">
          <i class="fa fa-google-plus"></i><span>Share on Google&#43;</span>
        </a>
      </li>
    
  </ul>
</div>
<div id="share-options-mask" class="share-options-mask"></div>
    </div>
    
    <div id="about">
  <div id="about-card">
    <div id="about-btn-close">
      <i class="fa fa-remove"></i>
    </div>
    
      <img id="about-card-picture" src="https://i.loli.net/2018/02/26/5a937bb148b02.jpg" alt="Author&#39;s picture" />
    
    <h4 id="about-card-name">Jixing Liu</h4>
    
      <div id="about-card-bio">Reading And Writing</div>
    
    
      <div id="about-card-job">
        <i class="fa fa-briefcase"></i>
        <br/>
        Data Scientist
      </div>
    
    
      <div id="about-card-location">
        <i class="fa fa-map-marker"></i>
        <br/>
        China
      </div>
    
  </div>
</div>

    <div id="algolia-search-modal" class="modal-container">
  <div class="modal">
    <div class="modal-header">
      <span class="close-button"><i class="fa fa-close"></i></span>
      <a href="https://algolia.com" target="_blank" rel="noopener" class="searchby-algolia text-color-light link-unstyled">
        <span class="searchby-algolia-text text-color-light text-small">by</span>
        <img class="searchby-algolia-logo" src="https://www.algolia.com/static_assets/images/press/downloads/algolia-light.svg">
      </a>
      <i class="search-icon fa fa-search"></i>
      <form id="algolia-search-form">
        <input type="text" id="algolia-search-input" name="search"
          class="form-control input--large search-input" placeholder="Search" />
      </form>
    </div>
    <div class="modal-body">
      <div class="no-result text-color-light text-center">no post found</div>
      <div class="results">
        
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="/2018/07/missing-data-imputation-and-fit-regression/">
                <h3 class="media-heading">Missing data imputation and fit regression</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Jul 7, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather">Mind map  Goal impute missing data fit regression  library(tidyverse) library(AER) library(naniar) library(mice)  Steps Step 01. missing data wages_data &lt;- read_csv(&quot;/Users/zero/myrepo/jixingBlogdown/data/Mroz.csv&quot;) wages_data &lt;- wages_data %&gt;% select(wage, educ, fatheduc, motheduc, inlf, hours, kidslt6, kidsge6, age, huswage, mtr, unem, city, exper) %&gt;% mutate_at(vars(kidslt6, kidsge6, hours, educ, age, wage, huswage, mtr, motheduc, fatheduc, unem, exper), as.numeric) %&gt;% mutate_at(vars(city, inlf), as.character) wages_data &lt;- wages_data %&gt;% mutate(wage = ifelse(is.na(wage), 0, wage)) vis_miss(wages_data) wages_mis &lt;- ampute(wages_data)$amp vis_miss(wages_mis)  Step 02.</div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="/2018/07/biostrings/">
                <h3 class="media-heading">ğŸ“¦ BioStrings</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Jul 7, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather">Init  DNA # DNA x &lt;- DNAString(&quot;actttGtag&quot;) is(x) ## [1] &quot;DNAString&quot; &quot;XString&quot; &quot;XRaw&quot; &quot;XVector&quot; &quot;Vector&quot; &quot;Annotated&quot; #è½¬å½• x %&gt;% complement() %&gt;% RNAString() -&gt; rna # åå‘äº’è¡¥ x.r &lt;- reverseComplement(x) x.r ## 9-letter &quot;DNAString&quot; instance ## seq: CTACAAAGT d &lt;- DNAString(&quot;TTGAAAA-CTC-N&quot;) d ## 13-letter &quot;DNAString&quot; instance ## seq: TTGAAAA-CTC-N #subset subseq(x, start =1, end = 5) ## 5-letter &quot;DNAString&quot; instance ## seq: ACTTT #ç¢±åŸºç½®æ¢ chartr(old = &quot;A&quot;, new = &quot;C&quot;, x) ## 9-letter &quot;DNAString&quot; instance ## seq: CCTTTGTCG #DNA_BASES DNA_BASES ## [1] &quot;A&quot; &quot;C&quot; &quot;G&quot; &quot;T&quot; #IUPAC_CODE_MAP IUPAC_CODE_MAP ## A C G T M R W S Y K ## &quot;A&quot; &quot;C&quot; &quot;G&quot; &quot;T&quot; &quot;AC&quot; &quot;AG&quot; &quot;AT&quot; &quot;CG&quot; &quot;CT&quot; &quot;GT&quot; ## V H D B N ## &quot;ACG&quot; &quot;ACT&quot; &quot;AGT&quot; &quot;CGT&quot; &quot;ACGT&quot;  RNA x &lt;- RNAString(&quot;acuuuGuag&quot;) is(x) ## [1] &quot;RNAString&quot; &quot;XString&quot; &quot;XRaw&quot; &quot;XVector&quot; &quot;Vector&quot; &quot;Annotated&quot; # é€†è½¬å½• x %&gt;% complement() %&gt;% DNAString() ## 9-letter &quot;DNAString&quot; instance ## seq: TGAAACATC # ç¿»è¯‘ codons(x) ## Views on a 9-letter RNAString subject ## subject: ACUUUGUAG ## views: ## start end width ## [1] 1 3 3 [ACU] ## [2] 4 6 3 [UUG] ## [3] 7 9 3 [UAG] translate(x) ## 3-letter &quot;AAString&quot; instance ## seq: TL* #dna -&gt; rna:T &lt;-&gt; U DNAString(&quot;actttGtag&quot;) %&gt;% RNAString() ## 9-letter &quot;RNAString&quot; instance ## seq: ACUUUGUAG #rna -&gt; dna:U &lt;-&gt; T rna %&gt;% DNAString() ## 9-letter &quot;DNAString&quot; instance ## seq: TGAAACATC  AA: Amino acid x &lt;- AAString(&quot;actttGtag&quot;) is(x) ## [1] &quot;AAString&quot; &quot;XString&quot; &quot;XRaw&quot; &quot;XVector&quot; &quot;Vector&quot; &quot;Annotated&quot;  reads # all kmer reads &lt;- mkAllStrings(c(&quot;A&quot;, &quot;C&quot;, &quot;G&quot;, &quot;T&quot;), 6) is(reads) ## [1] &quot;character&quot; &quot;vector&quot; &quot;data.</div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="/2018/05/purrr-use-partial-and-map-instead-of-loop/">
                <h3 class="media-heading">purrr: Use partial and map instead of loop</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  May 5, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather">ğŸŒ I want to â€¦   create three normal distribution with purrr::map as simple as possible
  ğŸŒ Hereâ€™s how to:   library(tidyverse) #rnorm(n=5, mean = mu, sd = sigma) #å‡½æ•°çš„ä¸€èˆ¬å½¢å¼ black_box &lt;- partial(rnorm, n=5)# å…ˆæŠŠä¸å˜çš„å›ºå®šä¸‹æ¥ï¼Œç”Ÿæˆæ–°çš„é»‘ç®±å‡½æ•°ï¼Œé»‘ç®±å‡½æ•°åªéœ€è¦æ¥å— map ä¼ é€’çš„å‚æ•°å³å¯ï¼Œ å…¶ä»–çš„å‚æ•°å·²ç»é€šè¿‡ Â·partialÂ· å›ºå®šä¸‹æ¥äº† mu &lt;- list(10, 100, -100)# å‚æ•° x sigma &lt;- list(0.01, 1, 10)# å‚æ•° y map2(mu, sigma, ~ black_box(mean=.x, sd=.y)) %&gt;% do.call(cbind, .) ## [,1] [,2] [,3] ## [1,] 10.</div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="/2018/05/iga-three-subgroup-deg-and-enrichment-analysis/">
                <h3 class="media-heading">IgA-three-subgroup-DEG-and-enrichment-analysis</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  May 5, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather">Enrichment analysis is a statistical procedure used to identify biological terms which are over-represented in a given gene set. These include signaling pathways , molecular functions, diseases, and a wide variety of other biological terms obtained by integrating prior knowledge of gene function from multiple resources.
Enrichr links : links to Enrichr containing the results of enrichment analyses generated by analyzing the up-regulated and down-regulated genes from a differential expression analysis.</div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="/2018/05/lelts-speaking-constructing-your-answers/">
                <h3 class="media-heading">LELTS speaking: constructing your answers</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  May 5, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather">å››ä¸ªè¦æ³¨æ„çš„é—®é¢˜ 
 æµç•…æ€§ ä¸°å¯Œå¤šå˜çš„è¯æ±‡ è¯­æ³•å’Œå‡†ç¡® å‘éŸ³   ç­”æ¡ˆçš„åŸºæœ¬ç»“æ„ 
ç›´æ¥å›ç­” + è¿æ¥è¯ + æ‰©å±•å¥å­ ï¼š
 åŸå›  ç»†èŠ‚ ä¾‹å­ äº‹æƒ…çš„ä¸¤é¢æ€§   å¸¸è§è¯è¯­çš„æ›¿ä»£ 
 And
 in addition to what i have just mentioned earlier on the top of that as well as this second to that among other things as a further matter   So
 this is the reason why this wxplains why and the result of this is   But</div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="/2018/05/r-vs-python-representing-data-in-r-python-equivalent/">
                <h3 class="media-heading">R vs python</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  May 5, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather">Representing Data in R &ndash; Python equivalent import pandas as pd import numpy as np  # 'characters' is equivalent to string firstName = 'jeff' print((type(firstName), firstName))  &lt;type 'str'&gt; jeff  # 'numeric' is equivalent to float heightCM = 188.2 print((type(heightCM), heightCM))  &lt;type 'float'&gt; 188.2  # integer is equivalent to integer numberSons = 1 print((type(numberSons), numberSons))  &lt;type 'int'&gt; 1  # 'logical' is equivalent to Boolean teachingCoursera = True print((type(teachingCoursera), teachingCoursera))  &lt;type 'bool'&gt; True  # 'vectors' is equivalent to numpy array or Python list (I will use array everywhere for consistency) heights = np.</div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="/2018/04/how-to-get-genomewide-position-specific-scores/">
                <h3 class="media-heading">How To Get Genomewide Position-Specific Scores</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Apr 4, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather">load packages #source(&quot;http://bioconductor.org/biocLite.R&quot;) #biocLite(&quot;GenomicScores&quot;) library(GenomicScores)  Retrieval of genomic scores through annotation packages There are currently four different annotation packages that store genomic scores and can be accessed using theÂ GenomicScoresÂ package
Annotation packages  Description 1. phastCons100way.UCSC.hg19 phastCons scores derived from the alignment of the human genome (hg19) to other 99 vertebrate species. 2. phastCons100way.UCSC.hg38 phastCons scores derived from the alignment of the human genome (hg38) to other 99 vertebrate species.</div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="/2018/04/purrr-adverb/">
                <h3 class="media-heading">purrr: adverb</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Apr 4, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather">What is an adverb Read carefully theÂ purrr documentation
 Adverbs modify the action of a function ; taking a function as input and returning a function with modified action as output.
 In other words, adverbs take a function, and return this function modified. Yes, just an adverb modifies a verb.
library(purrr) safe_log &lt;- safely(log)#high-order functions safe_log(&quot;a&quot;) ## $result ## NULL ## ## $error ## &lt;simpleError in log(x = x, base = base): non-numeric argument to mathematical function&gt; # have a result and error  how to write your own?</div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="/2018/04/why-did-my-plan-fail-again-2018-04-20/">
                <h3 class="media-heading">ä¸ºä»€ä¹ˆæˆ‘çš„é”»ç‚¼åˆå¤±è´¥äº†</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Apr 4, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather">  å¦ˆè›‹ï¼   å¥åº·é¥®é£Ÿ
åšæŒé”»ç‚¼ 
å¥½å¤„å¤šå¤š
äººå°½çš†çŸ¥
ä½†è¦çœŸæ­£åšåˆ°è¿™ä¸¤ç‚¹ çœŸå¿ƒä¸å®¹æ˜“
ä»Šå¤©å°±å¥½å¥½åæ€ä¸€ä¸‹ æˆ‘ä¸ºä»€ä¹ˆä¸èƒ½åšæŒé”»ç‚¼
æ”¾å¼ƒçš„åŸå› : . æŒ‡å®šçš„è®¡åˆ’éš¾åº¦å¤ªé«˜ . çŸ­æœŸæ²¡çœ‹åˆ°æ•ˆæœå°±æ˜¯å»äº†åŠ¨åŠ› . è¢«å„ç§çäº‹å¹²æ‰°ï¼Œæ— æ³•æ­£å¸¸æ‰§è¡Œ . åªè¦ä¸­é€”æ–­äº†å‡ æ¬¡ï¼Œå°±ä¼šæ”¾å¼ƒé”»ç‚¼çš„ä¹ æƒ¯  èƒ½è®©äººåœ¨çŸ­æœŸå†…å–å¾—æ˜¾è‘—æˆæ•ˆçš„è®¡åˆ’ï¼Œä½ éƒ½è¦ç‰¹åˆ«å°å¿ƒã€‚å› ä¸ºå¾€å¾€è¿™ç§è®¡åˆ’å¤ªè¿‡æ¿€çƒˆï¼Œæ²¡äººèƒ½é•¿æœŸåšæŒï¼Œå¾ˆå¿«ä½ å°±ä¼šæ”¾å¼ƒï¼Œå–å¾—çš„æˆæ•ˆä¹Ÿä¼šéšå®ƒè€Œå»ï¼Œç”šè‡³è¿˜æœ‰åå¼¹æ•ˆæœ
  è§£å†³åŠæ³•: æŠŠé”»ç‚¼å˜æˆåƒåˆ·ç‰™ä¸€æ ·çš„å¸¸è§„æ´»åŠ¨ï¼Œé›†ä¸­ç²¾åŠ›æƒ³å°½ä¸€åˆ‡åŠæ³•æŠŠå®ƒè¾¹åšå¸¸è§„æ´»åŠ¨ . ä»å°çš„è¿åŠ¨å¼€å§‹, å°½é‡ä»å°çš„å¼€å§‹ç›´åˆ°ä½ å­¦ä¼šäº†åšæŒã€‚å¯ä»¥æ…¢æ…¢å¢åŠ è¿åŠ¨é‡ï¼Œå“ªæ€•ä½ ä¸€å¼€å§‹èƒ½å¤Ÿå¤šåšä¸€äº›ï¼Œä¹Ÿä¸€å®šè¦å¿ä½ï¼Œ . å®‰æ’å›ºå®šçš„é”»ç‚¼æ—¶é—´ï¼Œè¿™æ®µæ—¶é—´æœ€å¥½èƒ½ä¸å—åˆ°å…¶ä»–äº‹æƒ…çš„å¹²æ‰° . å…»æˆä¹ æƒ¯ä¸­æœ€é‡è¦çš„ä¸€ç‚¹å°±æ˜¯åšæŒä¸æ‡ˆï¼Œåªè¦é”™è¿‡å‡ æ¬¡ï¼Œä½ å°±ä¼šå¼€å§‹çŠ¯æ‡’ã€‚å¤šå°‘æ¬¡éƒ½æ˜¯å› ä¸ºè¿™ä¸ªè´¥ä¸‹é˜µæ¥ï¼ŒğŸ˜” . æœ€å¥½æ‰¾ä¸ªæ­æ¡£ï¼Œå’Œåˆ«äººçº¦å®šä¸€èµ·ï¼Œä¼šå¸®åŠ©ä½ è‡ªå¾‹  </div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="/2018/04/algorithms-quick-sort/">
                <h3 class="media-heading">algorithms: quick sort</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Apr 4, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather">å¿«é€Ÿæ’åº å’Œ åˆ†è€Œæ²»ä¹‹ï¼ˆD&amp;Cï¼‰divde and conquer  D&amp;C å°†é—®é¢˜é€æ­¥åˆ†è§£ï¼Œæ‰¾åˆ°æœ€åŸºæœ¬çš„æƒ…å†µä¸ºæ­¢ å¿«é€Ÿæ’åºï¼Œéšæœºçš„é€‰æ‹©åŸºå‡†å€¼ï¼Œå¹³å‡è¿è¡Œæ—¶é—´ä¸º**O(n*logn)**  def quicksort(array): if len(array) &lt; 2:# base case, arrays with 0 or 1 element are already &quot;sorted&quot; return array else:# recursive case pivot = array[0]# sub-array of all the elements less than the pivot less = [i for i in array[1:] if i &lt;= pivot]# sub-array of all the elements greater than the pivot greater = [i for i in array[1:] if i &gt; pivot] return quicksort(less) + [pivot] + quicksort(greater) print(quicksort([10, 5, 2, 3]))  ## [2, 3, 5, 10]  </div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
      </div>
    </div>
    <div class="modal-footer">
      <p class="results-count text-medium"
         data-message-zero="no post found"
         data-message-one="1 post found"
         data-message-other="{n} posts found">
         34 posts found
      </p>
    </div>
  </div>
</div>
    
  
    
    <div id="cover" style="background-image:url('https://i.loli.net/2018/02/26/5a9416186c149.jpg');"></div>
  


    
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.1.3/jquery.min.js" integrity="sha256-IFHWFEbU2/+wNycDECKgjIRSirRNIDp2acEB5fvdVRU=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.8.0/highlight.min.js" integrity="sha256-+mpyNVJsNt4rVXCw0F+pAOiB3YxmHgrbJsx4ecPuUaI=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.4/jquery.fancybox.min.js" integrity="sha256-vMxgR/7FtLovVA+IPrR7+xTgIgARH7y9VZQnmmi0HDI=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.4/helpers/jquery.fancybox-thumbs.min.js" integrity="sha256-N0qFUh7/9vLvia87dDndewmsgsyYoNkdA212tPc+2NI=" crossorigin="anonymous"></script>


<script src="/js/script-kr8dyqj6rb6ortyib7whfo9x9p6td6zo8t1v4fdz4ecx5kwybsdlmk1slygn.min.js"></script>


<script>
$(document).ready(function() {
  hljs.configure({ classPrefix: '', useBR: false });
  $('pre.code-highlight, pre > code').each(function(i, block) {
    if (!$(this).hasClass('codeblock')) {
      $(this).addClass('codeblock');
    }
    hljs.highlightBlock(block);
  });
});
</script>

  
    
      <script>
        var disqus_config = function () {
          this.page.url = '\/2018\/04\/how-to-get-genomewide-position-specific-scores\/';
          
            this.page.identifier = '\/2018\/04\/how-to-get-genomewide-position-specific-scores\/'
          
        };
        (function() {
          
          
          if (window.location.hostname == "localhost") {
            return;
          }
          var d = document, s = d.createElement('script');
          var disqus_shortname = 'jixing';
          s.src = '//' + disqus_shortname + '.disqus.com/embed.js';

          s.setAttribute('data-timestamp', +new Date());
          (d.head || d.body).appendChild(s);
        })();
      </script>
    
  



    
  </body>
</html>

