

  
    
  


  





  

<!DOCTYPE html>
<html lang="en-us">
  <head>
    
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="generator" content="Hugo 0.32.2 with theme Tranquilpeak 0.4.1-BETA">
    <title>How To Get Genomewide Position-Specific Scores</title>
    <meta name="author" content="Jixing Liu">
    <meta name="keywords" content=", R">

    <link rel="icon" href="img/R_py.png">
    

    
    <meta name="description" content="load packages #source(&quot;http://bioconductor.org/biocLite.R&quot;) #biocLite(&quot;GenomicScores&quot;) library(GenomicScores)  Retrieval of genomic scores through annotation packages There are currently four different annotation packages that store genomic scores and can be accessed using theÂ GenomicScoresÂ package
Annotation packages  Description 1. phastCons100way.UCSC.hg19 phastCons scores derived from the alignment of the human genome (hg19) to other 99 vertebrate species. 2. phastCons100way.UCSC.hg38 phastCons scores derived from the alignment of the human genome (hg38) to other 99 vertebrate species.">
    <meta property="og:description" content="load packages #source(&quot;http://bioconductor.org/biocLite.R&quot;) #biocLite(&quot;GenomicScores&quot;) library(GenomicScores)  Retrieval of genomic scores through annotation packages There are currently four different annotation packages that store genomic scores and can be accessed using theÂ GenomicScoresÂ package
Annotation packages  Description 1. phastCons100way.UCSC.hg19 phastCons scores derived from the alignment of the human genome (hg19) to other 99 vertebrate species. 2. phastCons100way.UCSC.hg38 phastCons scores derived from the alignment of the human genome (hg38) to other 99 vertebrate species.">
    <meta property="og:type" content="blog">
    <meta property="og:title" content="How To Get Genomewide Position-Specific Scores">
    <meta property="og:url" content="/2018/04/how-to-get-genomewide-position-specific-scores/">
    <meta property="og:site_name" content="student zero ">
    <meta name="twitter:card" content="summary">
    <meta name="twitter:title" content="student zero ">
    <meta name="twitter:description" content="load packages #source(&quot;http://bioconductor.org/biocLite.R&quot;) #biocLite(&quot;GenomicScores&quot;) library(GenomicScores)  Retrieval of genomic scores through annotation packages There are currently four different annotation packages that store genomic scores and can be accessed using theÂ GenomicScoresÂ package
Annotation packages  Description 1. phastCons100way.UCSC.hg19 phastCons scores derived from the alignment of the human genome (hg19) to other 99 vertebrate species. 2. phastCons100way.UCSC.hg38 phastCons scores derived from the alignment of the human genome (hg38) to other 99 vertebrate species.">
    
      <meta name="twitter:creator" content="@studentZero475">
    
    

    
    

    
      <meta property="og:image" content="https://i.loli.net/2018/02/26/5a937bb148b02.jpg">
    

    
      <meta property="og:image" content="/post/2018-04-27-how-to-get-genomewide-position-specific-scores_files/download.jpg">
    
    
    

    

    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.4/jquery.fancybox.min.css" integrity="sha256-vuXZ9LGmmwtjqFX1F+EKin1ThZMub58gKULUyf0qECk=" crossorigin="anonymous" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.4/helpers/jquery.fancybox-thumbs.min.css" integrity="sha256-SEa4XYAHihTcEP1f5gARTB2K26Uk8PsndQYHQC1f4jU=" crossorigin="anonymous" />
    
    
    <link rel="stylesheet" href="/css/style-hbs5om8csx9a8yrv5hnhpi2qqdv4ykuslajwbramhqxvleqbfklxgek50hye.min.css" />
    
    

    
      
    
    
  </head>

  <body>
    <div id="blog">
      <link rel="stylesheet" href="/css/monokai-sublime.css" rel="stylesheet" id="theme-stylesheet">
<script src="/js/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>

<header id="header" data-behavior="4">
  <i id="btn-open-sidebar" class="fa fa-lg fa-bars"></i>
  <div class="header-title">
    <a class="header-title-link" href="/">student zero </a>
  </div>
  
    
      <a class="header-right-picture "
         href="/#about">
    
    
    
      
        <img class="header-picture" src="https://i.loli.net/2018/02/26/5a937bb148b02.jpg" alt="Author&#39;s picture" />
      
    
    </a>
  
</header>

      <nav id="sidebar" data-behavior="4">
  <div class="sidebar-container">
    
      <div class="sidebar-profile">
        <h1 class="sidebar-profile-title">How To Get Genomewide Position-Specific Scores</h1>
        <a href="/#about">
          <img class="sidebar-profile-picture" src="https://i.loli.net/2018/02/26/5a937bb148b02.jpg" alt="Author&#39;s picture" />
        </a>
        <h4 class="sidebar-profile-name">Jixing Liu</h4>
        
          <h5 class="sidebar-profile-bio">Reading And Writing</h5>
        
      </div>
    
    <ul class="sidebar-buttons">
      
  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="/">
    
      <i class="sidebar-button-icon fa fa-lg fa-home"></i>
      
      <span class="sidebar-button-desc">Home</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="/archives/">
    
      <i class="sidebar-button-icon fa fa-lg fa-archive"></i>
      
      <span class="sidebar-button-desc">Archives</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="/categories/">
    
      <i class="sidebar-button-icon fa fa-lg fa-bookmark"></i>
      
      <span class="sidebar-button-desc">Categories</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="/tags/">
    
      <i class="sidebar-button-icon fa fa-lg fa-tags"></i>
      
      <span class="sidebar-button-desc">Tags</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="/page/about/">
    
      <i class="sidebar-button-icon fa fa-lg fa-question"></i>
      
      <span class="sidebar-button-desc">About</span>
    </a>
  </li>


    </ul>
    <ul class="sidebar-buttons">
      
  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://twitter.com/studentZero475">
    
      <i class="sidebar-button-icon fa fa-lg fa-twitter"></i>
      
      <span class="sidebar-button-desc">Twitter</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://www.r-bloggers.com/">
    
      <i class="sidebar-button-icon fa fa-lg fa-book"></i>
      
      <span class="sidebar-button-desc">R-bloggers</span>
    </a>
  </li>


    </ul>
    <ul class="sidebar-buttons">
      

    </ul>
  </div>
</nav>

      

      <div id="main" data-behavior="4"
        class="
               hasCoverMetaOut
               ">
        <article class="post" itemscope itemType="http://schema.org/BlogPosting">
          
          
            <div class="post-header main-content-wrap text-center">
  
    <h1 class="post-title" itemprop="headline">
      How To Get Genomewide Position-Specific Scores
    </h1>
  
  
  <div class="postShorten-meta post-meta">
    
      <time itemprop="datePublished" datetime="2018-04-27T00:00:00Z">
        
  April 27, 2018

      </time>
    
    
  
  
    <span>in</span>
    
      <a class="category-link" href="/categories/bioinformatics">bioinformatics</a>
    
  


  </div>

</div>
          
          <div class="post-content markdown" itemprop="articleBody">
            <div class="main-content-wrap">
              <div id="load-packages" class="section level2">
<h2>load packages</h2>
<pre class="r"><code>#source(&quot;http://bioconductor.org/biocLite.R&quot;)
#biocLite(&quot;GenomicScores&quot;)
library(GenomicScores)</code></pre>
</div>
<div id="retrieval-of-genomic-scores-through-annotation-packages" class="section level2">
<h2>Retrieval of genomic scores through annotation packages</h2>
<p>There are currently four different annotation packages that store genomic scores and can be accessed using theÂ <em><a href="http://bioconductor.org/packages/GenomicScores">GenomicScores</a></em>Â package</p>
<p><strong style="color: darkred;">Annotation packages </strong> <strong style="color: darkred;">Description</strong> 1. <em><a href="http://bioconductor.org/packages/phastCons100way.UCSC.hg19">phastCons100way.UCSC.hg19</a></em> phastCons scores derived from the alignment of the human genome (hg19) to other 99 vertebrate species. 2. <em><a href="http://bioconductor.org/packages/phastCons100way.UCSC.hg38">phastCons100way.UCSC.hg38</a></em> phastCons scores derived from the alignment of the human genome (hg38) to other 99 vertebrate species. 3. <em><a href="http://bioconductor.org/packages/phastCons7way.UCSC.hg38">phastCons7way.UCSC.hg38</a></em> phastCons scores derived from the alignment of the human genome (hg38) to other 6 mammal species. 4. <em><a href="http://bioconductor.org/packages/fitCons.UCSC.hg19">fitCons.UCSC.hg19</a></em> fitCons scores: fitness consequences of functional annotation for the human genome (hg19).</p>
<div id="to-retrieve-genomic-scores-for-specific-positions-we-should-use-the-function-scores" class="section level3">
<h3><strong style="color: darkred;">To retrieve genomic scores for specific positions we should use the function scores()</strong></h3>
<pre class="r"><code>library(phastCons100way.UCSC.hg19)
gsco &lt;- phastCons100way.UCSC.hg19
class(gsco)</code></pre>
<pre><code>## [1] &quot;GScores&quot;
## attr(,&quot;package&quot;)
## [1] &quot;GenomicScores&quot;</code></pre>
<pre class="r"><code>scores(gsco, GRanges(seqnames=&quot;chr22&quot;,
                     IRanges(start=50967020:50967021, width=1)))</code></pre>
<pre><code>## GRanges object with 2 ranges and 1 metadata column:
##       seqnames               ranges strand |    scores
##          &lt;Rle&gt;            &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
##   [1]    chr22 [50967020, 50967020]      * |         1
##   [2]    chr22 [50967021, 50967021]      * |         1
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<pre class="r"><code>gsco</code></pre>
<pre><code>## GScores object 
## # organism: Homo sapiens (UCSC)
## # provider: UCSC
## # provider version: 09Feb2014
## # download date: Mar 17, 2017
## # loaded sequences: chr19_gl000208_random, chr22
## # maximum abs. error: 0.05</code></pre>
<pre class="r"><code>#citation(gsco) now cann&#39;t run
provider(gsco)</code></pre>
<pre><code>## [1] &quot;UCSC&quot;</code></pre>
<pre class="r"><code>providerVersion(gsco)</code></pre>
<pre><code>## [1] &quot;09Feb2014&quot;</code></pre>
<pre class="r"><code>organism(gsco)</code></pre>
<pre><code>## [1] &quot;Homo sapiens&quot;</code></pre>
<pre class="r"><code>seqlevelsStyle(gsco)</code></pre>
<pre><code>## [1] &quot;UCSC&quot;</code></pre>
</div>
</div>
<div id="retrieval-of-genomic-scores-through-annotationhub-resources" class="section level2">
<h2>Retrieval of genomic scores through <strong style="color: darkred;">AnnotationHub resources</strong></h2>
<p>Another way to retrieve genomic scores is by using theÂ <em><a href="http://bioconductor.org/packages/AnnotationHub">AnnotationHub</a></em>, which is a web resource that provides a central location where genomic files (e.g., VCF, bed, wig) and other resources from standard (e.g., UCSC, Ensembl) and distributed sites, can be found. A BioconductorÂ <em><a href="http://bioconductor.org/packages/AnnotationHub">AnnotationHub</a></em>Â web resource creates and manages a local cache of files retrieved by the user, helping with quick and reproducible access.</p>
<p><img src="/post/2018-04-27-how-to-get-genomewide-position-specific-scores_files/Screen Shot 2018-04-27 at 15.35.21.png" width="30%" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gsco &lt;- getGScores(&quot;phastCons7way.UCSC.hg38&quot;)
scores(gsco, GRanges(seqnames=&quot;chr22&quot;,IRanges(start=20967020:20967025, width=1)))</code></pre>
</div>
<div id="retrieval-of-multiple-scores-per-genomic-position" class="section level2">
<h2>Retrieval of <strong style="color: darkred;">multiple scores</strong> per genomic position</h2>
</div>
<div id="summarization-of-genomic-scores" class="section level2">
<h2>Summarization of genomic scores</h2>
<pre class="r"><code>library(dplyr)
gsco &lt;- phastCons100way.UCSC.hg19
gr &lt;- GRanges(seqnames=&quot;chr22&quot;, IRanges(start=50967020:50967025, width=1))

#mean
scores(gsco, gr) %&gt;% 
  .$scores %&gt;% 
  mean()</code></pre>
<pre><code>## [1] 0.8</code></pre>
<pre class="r"><code>#median
scores(gsco, gr) %&gt;% 
  .$scores %&gt;% 
  median()</code></pre>
<pre><code>## [1] 1</code></pre>
<pre class="r"><code>#min
scores(gsco, gr) %&gt;% 
  .$scores %&gt;% 
  min()</code></pre>
<pre><code>## [1] 0</code></pre>
<pre class="r"><code>scores(gsco, gr) %&gt;% 
  .$scores %&gt;% 
  hist()</code></pre>
<p><img src="/post/2018-04-27-how-to-get-genomewide-position-specific-scores_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
</div>
<div id="annotating-variants-with-genomic-scores" class="section level2">
<h2><strong style="color: darkred;">Annotating variants with genomic scores</strong></h2>
<p>A typical use case of theÂ <em><a href="http://bioconductor.org/packages/GenomicScores">GenomicScores</a></em>Â package is in the context of annotating variants with genomic scores, such as phastCons conservation scores. For this purpose, we load theÂ <em><a href="http://bioconductor.org/packages/VariantAnnotaiton">VariantAnnotaiton</a></em>Â andÂ <em><a href="http://bioconductor.org/packages/TxDb.Hsapiens.UCSC.hg19.knownGene">TxDb.Hsapiens.UCSC.hg19.knownGene</a></em>Â packages. The former will allow us to read a VCF file and annotate it, and the latter contains the gene annotations from UCSC that will be used in this process.</p>
<p>Letâ€™s load one of the sample VCF files that form part of theÂ <em><a href="http://bioconductor.org/packages/VariantAnnotation">VariantAnnotation</a></em>Â package.</p>
<pre class="r"><code>library(VariantAnnotation)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
fl &lt;- system.file(&quot;extdata&quot;, &quot;chr22.vcf.gz&quot;, package=&quot;VariantAnnotation&quot;)
vcf &lt;- readVcf(fl, &quot;hg19&quot;)
seqlevelsStyle(vcf)</code></pre>
<pre><code>## [1] &quot;NCBI&quot;    &quot;Ensembl&quot;</code></pre>
<pre class="r"><code>txdb &lt;- TxDb.Hsapiens.UCSC.hg19.knownGene
seqlevelsStyle(txdb)</code></pre>
<pre><code>## [1] &quot;UCSC&quot;</code></pre>
<p>Because the chromosome nomenclature from the VCF file (NCBI) is different from the one with the gene annotations (UCSC) we use the <strong style="color: darkred;">seqlevelsStyle()</strong> function to force our variants having the chromosome nomenclature of the gene annotations.</p>
<pre class="r"><code>seqlevelsStyle(vcf) &lt;- seqlevelsStyle(txdb)</code></pre>
<p>We <strong style="color: darkred;">annotate the location of variants</strong> using the function <strong style="color: darkred;">locateVariants()</strong> from the VariantAnnotation package.</p>
<pre class="r"><code>loc &lt;- locateVariants(vcf, txdb, AllVariants())
loc[1:3]</code></pre>
<pre><code>## GRanges object with 3 ranges and 9 metadata columns:
##    seqnames               ranges strand | LOCATION  LOCSTART    LOCEND
##       &lt;Rle&gt;            &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt; &lt;integer&gt; &lt;integer&gt;
##       chr22 [50300078, 50300078]      - |   intron     10763     10763
##       chr22 [50300086, 50300086]      - |   intron     10755     10755
##       chr22 [50300101, 50300101]      - |   intron     10740     10740
##      QUERYID        TXID         CDSID      GENEID       PRECEDEID
##    &lt;integer&gt; &lt;character&gt; &lt;IntegerList&gt; &lt;character&gt; &lt;CharacterList&gt;
##            1       75253                     79087                
##            2       75253                     79087                
##            3       75253                     79087                
##           FOLLOWID
##    &lt;CharacterList&gt;
##                   
##                   
##                   
##   -------
##   seqinfo: 1 sequence from hg19 genome; no seqlengths</code></pre>
<pre class="r"><code>table(loc$LOCATION)</code></pre>
<pre><code>## 
## spliceSite     intron    fiveUTR   threeUTR     coding intergenic 
##         11      22572        309       1368       2822       2867 
##   promoter 
##       2864</code></pre>
</div>
<div id="annotate-phastcons-conservation-scores" class="section level2">
<h2><strong style="color: darkred;">Annotate phastCons conservation scores</strong></h2>
<p>on the variants and store those annotations as an additional metadata column of the GRanges object. For this specific purpose we should the argument scores.only=TRUE that makes the scores() method to return the genomic scores as a numeric vector instead as a metadata column in the input ranges object.</p>
<pre class="r"><code>loc$PHASTCONS &lt;- scores(gsco, loc, scores.only=TRUE)
loc[1:3]</code></pre>
<pre><code>## GRanges object with 3 ranges and 10 metadata columns:
##    seqnames               ranges strand | LOCATION  LOCSTART    LOCEND
##       &lt;Rle&gt;            &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt; &lt;integer&gt; &lt;integer&gt;
##       chr22 [50300078, 50300078]      - |   intron     10763     10763
##       chr22 [50300086, 50300086]      - |   intron     10755     10755
##       chr22 [50300101, 50300101]      - |   intron     10740     10740
##      QUERYID        TXID         CDSID      GENEID       PRECEDEID
##    &lt;integer&gt; &lt;character&gt; &lt;IntegerList&gt; &lt;character&gt; &lt;CharacterList&gt;
##            1       75253                     79087                
##            2       75253                     79087                
##            3       75253                     79087                
##           FOLLOWID PHASTCONS
##    &lt;CharacterList&gt; &lt;numeric&gt;
##                          0.0
##                          0.1
##                          0.0
##   -------
##   seqinfo: 1 sequence from hg19 genome; no seqlengths</code></pre>
<p>Using the following code we can examine <strong style="color: darkred;">the distribution of phastCons conservation scores</strong> of variants across the different annotated regions</p>
<pre class="r"><code>x &lt;- split(loc$PHASTCONS, loc$LOCATION)
mask &lt;- elementNROWS(x) &gt; 0
boxplot(x[mask], ylab=&quot;phastCons score&quot;, las=1, cex.axis=1.2, cex.lab=1.5, col=&quot;gray&quot;)
points(1:length(x[mask])+0.25, 
       sapply(x[mask], mean, na.rm=TRUE),
       pch=23, bg=&quot;red&quot;)</code></pre>
<p><img src="/post/2018-04-27-how-to-get-genomewide-position-specific-scores_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
</div>
<div id="links" class="section level2">
<h2><strong>Links</strong></h2>
<ul>
<li><a href="http://bioconductor.org/packages/release/bioc/vignettes/GenomicScores/inst/doc/GenomicScores.html">An introduction to the GenomicScores package</a></li>
</ul>
</div>

              
            </div>
          </div>
          <div id="post-footer" class="post-footer main-content-wrap">
            
              
                
                
                  <div class="post-footer-tags">
                    <span class="text-color-light text-small">TAGGED IN</span><br/>
                    
  <a class="tag tag--primary tag--small" href="//tags/phast/">phast</a>

                  </div>
                
              
            
            
  <div class="post-actions-wrap">
      <nav >
        <ul class="post-actions post-action-nav">
            <li class="post-action">
              
                <a class="post-action-btn btn btn--default tooltip--top" href="/2018/05/r-vs-python-representing-data-in-r-python-equivalent/" data-tooltip="R vs python">
              
                <i class="fa fa-angle-left"></i>
                <span class="hide-xs hide-sm text-small icon-ml">NEXT</span>
              </a>
            </li>
            <li class="post-action">
              
                <a class="post-action-btn btn btn--default tooltip--top" href="/2018/04/purrr-adverb/" data-tooltip="purrr: adverb">
              
                <span class="hide-xs hide-sm text-small icon-mr">PREVIOUS</span>
                <i class="fa fa-angle-right"></i>
              </a>
            </li>
        </ul>
      </nav>
    <ul class="post-actions post-action-share" >
      
        <li class="post-action hide-lg hide-md hide-sm">
          <a class="post-action-btn btn btn--default btn-open-shareoptions" href="#btn-open-shareoptions">
            <i class="fa fa-share-alt"></i>
          </a>
        </li>
        
          <li class="post-action hide-xs">
            <a class="post-action-btn btn btn--default" target="new" href="https://www.facebook.com/sharer/sharer.php?u=/2018/04/how-to-get-genomewide-position-specific-scores/">
              <i class="fa fa-facebook-official"></i>
            </a>
          </li>
        
          <li class="post-action hide-xs">
            <a class="post-action-btn btn btn--default" target="new" href="https://twitter.com/intent/tweet?text=/2018/04/how-to-get-genomewide-position-specific-scores/">
              <i class="fa fa-twitter"></i>
            </a>
          </li>
        
          <li class="post-action hide-xs">
            <a class="post-action-btn btn btn--default" target="new" href="https://plus.google.com/share?url=/2018/04/how-to-get-genomewide-position-specific-scores/">
              <i class="fa fa-google-plus"></i>
            </a>
          </li>
        
      
      
        <li class="post-action">
          <a class="post-action-btn btn btn--default" href="#disqus_thread">
            <i class="fa fa-comment-o"></i>
          </a>
        </li>
      
      <li class="post-action">
        
          <a class="post-action-btn btn btn--default" href="#">
        
          <i class="fa fa-list"></i>
        </a>
      </li>
    </ul>
  </div>


            
              
                <div id="disqus_thread">
  <noscript>Please enable JavaScript to view the <a href="//disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
</div>
              
            
          </div>
        </article>
        <footer id="footer" class="main-content-wrap">
  <span class="copyrights">
    &copy; 2019 Jixing Liu. All Rights Reserved
  </span>
</footer>

      </div>
      <div id="bottom-bar" class="post-bottom-bar" data-behavior="4">
        
  <div class="post-actions-wrap">
      <nav >
        <ul class="post-actions post-action-nav">
            <li class="post-action">
              
                <a class="post-action-btn btn btn--default tooltip--top" href="/2018/05/r-vs-python-representing-data-in-r-python-equivalent/" data-tooltip="R vs python">
              
                <i class="fa fa-angle-left"></i>
                <span class="hide-xs hide-sm text-small icon-ml">NEXT</span>
              </a>
            </li>
            <li class="post-action">
              
                <a class="post-action-btn btn btn--default tooltip--top" href="/2018/04/purrr-adverb/" data-tooltip="purrr: adverb">
              
                <span class="hide-xs hide-sm text-small icon-mr">PREVIOUS</span>
                <i class="fa fa-angle-right"></i>
              </a>
            </li>
        </ul>
      </nav>
    <ul class="post-actions post-action-share" >
      
        <li class="post-action hide-lg hide-md hide-sm">
          <a class="post-action-btn btn btn--default btn-open-shareoptions" href="#btn-open-shareoptions">
            <i class="fa fa-share-alt"></i>
          </a>
        </li>
        
          <li class="post-action hide-xs">
            <a class="post-action-btn btn btn--default" target="new" href="https://www.facebook.com/sharer/sharer.php?u=/2018/04/how-to-get-genomewide-position-specific-scores/">
              <i class="fa fa-facebook-official"></i>
            </a>
          </li>
        
          <li class="post-action hide-xs">
            <a class="post-action-btn btn btn--default" target="new" href="https://twitter.com/intent/tweet?text=/2018/04/how-to-get-genomewide-position-specific-scores/">
              <i class="fa fa-twitter"></i>
            </a>
          </li>
        
          <li class="post-action hide-xs">
            <a class="post-action-btn btn btn--default" target="new" href="https://plus.google.com/share?url=/2018/04/how-to-get-genomewide-position-specific-scores/">
              <i class="fa fa-google-plus"></i>
            </a>
          </li>
        
      
      
        <li class="post-action">
          <a class="post-action-btn btn btn--default" href="#disqus_thread">
            <i class="fa fa-comment-o"></i>
          </a>
        </li>
      
      <li class="post-action">
        
          <a class="post-action-btn btn btn--default" href="#">
        
          <i class="fa fa-list"></i>
        </a>
      </li>
    </ul>
  </div>


      </div>
      <div id="share-options-bar" class="share-options-bar" data-behavior="4">
  <i id="btn-close-shareoptions" class="fa fa-close"></i>
  <ul class="share-options">
    
      <li class="share-option">
        <a class="share-option-btn" target="new" href="https://www.facebook.com/sharer/sharer.php?u=%2F2018%2F04%2Fhow-to-get-genomewide-position-specific-scores%2F">
          <i class="fa fa-facebook-official"></i><span>Share on Facebook</span>
        </a>
      </li>
    
      <li class="share-option">
        <a class="share-option-btn" target="new" href="https://twitter.com/intent/tweet?text=%2F2018%2F04%2Fhow-to-get-genomewide-position-specific-scores%2F">
          <i class="fa fa-twitter"></i><span>Share on Twitter</span>
        </a>
      </li>
    
      <li class="share-option">
        <a class="share-option-btn" target="new" href="https://plus.google.com/share?url=%2F2018%2F04%2Fhow-to-get-genomewide-position-specific-scores%2F">
          <i class="fa fa-google-plus"></i><span>Share on Google&#43;</span>
        </a>
      </li>
    
  </ul>
</div>
<div id="share-options-mask" class="share-options-mask"></div>
    </div>
    
    <div id="about">
  <div id="about-card">
    <div id="about-btn-close">
      <i class="fa fa-remove"></i>
    </div>
    
      <img id="about-card-picture" src="https://i.loli.net/2018/02/26/5a937bb148b02.jpg" alt="Author&#39;s picture" />
    
    <h4 id="about-card-name">Jixing Liu</h4>
    
      <div id="about-card-bio">Reading And Writing</div>
    
    
      <div id="about-card-job">
        <i class="fa fa-briefcase"></i>
        <br/>
        Data Scientist
      </div>
    
    
      <div id="about-card-location">
        <i class="fa fa-map-marker"></i>
        <br/>
        China
      </div>
    
  </div>
</div>

    <div id="algolia-search-modal" class="modal-container">
  <div class="modal">
    <div class="modal-header">
      <span class="close-button"><i class="fa fa-close"></i></span>
      <a href="https://algolia.com" target="_blank" rel="noopener" class="searchby-algolia text-color-light link-unstyled">
        <span class="searchby-algolia-text text-color-light text-small">by</span>
        <img class="searchby-algolia-logo" src="https://www.algolia.com/static_assets/images/press/downloads/algolia-light.svg">
      </a>
      <i class="search-icon fa fa-search"></i>
      <form id="algolia-search-form">
        <input type="text" id="algolia-search-input" name="search"
          class="form-control input--large search-input" placeholder="Search" />
      </form>
    </div>
    <div class="modal-body">
      <div class="no-result text-color-light text-center">no post found</div>
      <div class="results">
        
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="/2019/06/multi-label-classfication/">
                <h3 class="media-heading">å¤šæ ‡ç­¾åˆ†ç±»é—®é¢˜</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Jun 6, 2019
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather">Difference between multi-class classification &amp; multi-label classification is that in multi-class problems the classes are mutually exclusive, whereas for multi-label problems each label represents a different classification task, but the tasks are somehow related.
  multi-class classification makes the assumption that each sample is assigned to one and only one label: a fruit can be either an apple or a pear but not both at the same time.</div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="/2019/06/drug-discovery/">
                <h3 class="media-heading">æ–°è¯ç ”å‘</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Jun 6, 2019
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather">ç¡®å®šè‡´ç—…æœºåˆ¶ 1. çš‡åç¡®è¯Šä¹³è…ºç™Œ  2. è¯ç‰©éå¯»å‡æ— æ•ˆ  3. ç²¾å‡†åŒ»ç–—é¡¹ç›®å¯åŠ¨ ç”Ÿç‰©ä¿¡æ¯åˆ†æ ç‰¹å¼‚åŒ–çªå˜åŸºå› é‰´å®š è›‹ç™½é€šè·¯ä¸ç–¾ç—…å…³è”æ€§ XYZè›‹ç™½ä¸‰ç»´æ™¶ä½“ç»“æ„ XYZè›‹ç™½è¯ç‰©ç»“åˆå£è¢‹çªå˜å‰ååˆ†æ    å…ˆå¯¼åŒ–åˆç‰© 1. è®¡ç®—æœºè¾…åŠ©è¯ç‰©è®¾è®¡:  åœ¨ è›‹ç™½è´¨ä¸‰ç»´ç»“æ„åŸºç¡€ä¸Šï¼Œå½»åº•åˆ†æçªå˜åXYZè›‹ç™½ç»“åˆå£è¢‹å‘¨å›´ç†åŒ–æ€§è´¨
 çªå˜å‰ï¼šå°åˆ†å­åŒ…å›´åœ¨ç–æ°´å£è¢‹ï¼Œäº²æ°´æ°¨åŸºé…¸R125ã€E154ä¸å…¶å½¢æˆæ°¢é”®é“†é’‰ä½œç”¨ï¼Œä½¿å…¶ç‰¢ç‰¢ç¨³å›ºåœ¨è›‹ç™½è´¨ä¸­ çªå˜åï¼šç–æ°´ç¯å¢ƒå¤§è‡´ä¸å˜ï¼Œä½†æ°¨åŸºé…¸N125ã€I154(æ— æ³•å½¢æˆæ°¢é”®)é€ æˆå¾ˆå¤§ç©ºéš™ï¼Œäº²æ°´æ€§è½¬æ¢ä¸ºç–æ°´å£è¢‹ï¼Œç¨³å®šæ€§é™ä½     2. åˆ†å­åŠ¨åŠ›å­¦æ–¹æ³•  åˆ†å­åŠ¨åŠ›å­¦æ–¹æ³•å¯¹XYZè›‹ç™½çªå˜åä¸å°åˆ†å­Hormoneé—´ç›¸äº’ä½œç”¨çŠ¶å†µè¿›è¡Œäº†è®¡ç®—æ¨¡æ‹Ÿåˆ†æ, ç›®çš„:ä»¥æœŸå¾—åˆ°è›‹ç™½è´¨åœ¨çªå˜åçš„å°åˆ†å­ç»“åˆä¿¡æ¯
 è®¡ç®—ä¸­å¿ƒé¦–å¸­ç§‘å­¦å®¶æ˜¯è¿™ä¹ˆè§£é‡Šåˆ†æç»“æœçš„:
çªå˜åXYZè›‹ç™½Region AåŒºåŸŸæ²¡æœ‰å˜åŒ–ï¼Œå¯¹è¯¥éƒ¨åˆ†ä¸åšä»»ä½•ä¿®æ”¹ï¼› Region BåŒºåŸŸä¸­ï¼Œå°åˆ†å­ä¸‹æ–¹å‡ºç°å·¨å¤§ç©ºæ´ï¼Œéœ€è¦å¯¹è¯¥éƒ¨åˆ†è¿›è¡Œè¡¥æ¼ï¼› Region BåŒºåŸŸå·¨å¤§ç©ºè¢­ï¼Œåœ¨å°åˆ†å­ç¨³å®šç»“åˆæ—¶ï¼Œä¸‹æ–¹å‡ºç°ç”±10~13ä¸ªæ°´åˆ†å­ç»„æˆçš„â€œæ°´æ°¢é”®ç½‘ç»œä½“ç³»â€ï¼› â€œæ°´æ°¢é”®ç½‘ç»œä½“ç³»â€ä¸­ï¼Œæ°´åˆ†å­èƒ½é‡Energyï¼ˆçƒä½“å¤§å°ï¼‰å’Œå æœ‰ç‡Occupancyï¼ˆçƒä½“é¢œè‰²ï¼‰å‡èƒ½è¡¨è¾¾å‡ºè›‹ç™½è´¨å±€éƒ¨åŒºåŸŸç†åŒ–æ€§è´¨ã€‚ &gt; å…·ä½“æ€»ç»“ï¼šRegion BåŒºåŸŸç»å¤§å¤šæ•°æ°´åˆ†å­èƒ½é‡Energyæå°ã€å æœ‰ç‡Occupancyè¾ƒå°ï¼Œè¯´æ˜è¯¥åŒºåŸŸæä¸ºç–æ°´ï¼›ä½†å…¶ä¸­W1æ°´åˆ†å­ï¼Œèƒ½é‡è¾ƒå¥½ã€å æœ‰ç‡è¾ƒé«˜ï¼Œè¯´æ˜W1æ°´åˆ†å­æä¸ºç¨³å®šï¼Œå‘¨å›´æ°¨åŸºé…¸ä¸ºæäº²æ°´æ€§æ°¨åŸºé…¸ï¼Œå°†æ¥å¯ä»¥é€šè¿‡æ›¿æ¢W1æ°´åˆ†å­æˆ–é€šè¿‡æ¡¥é”®ä½œç”¨å¯¹å…¶è¿›è¡ŒåŒ–å­¦ä¿®é¥°  ; ;
 3. åŸºäºä½“å†…åŸç”Ÿæ¿€ç´ Hormoneçš„è¯ç‰©è®¾è®¡:åŒ–å­¦ç‰‡æ®µå¢é•¿è®¾è®¡æ³•  é¦–å…ˆè®¾è®¡çš„åŒ–å­¦ç‰‡æ®µï¼Œå°†èƒ½å¤Ÿæœ‰æ•ˆçš„åˆ°è¾¾W1æ°´åˆ†å­é™„è¿‘æˆ–å°†å…¶æ›¿æ¢æ‰, ç¦æ–¯å¦å›½ç«‹å¤§å­¦åŒ–å­¦é™¢ç ”ç©¶å›¢é˜Ÿï¼Œå…±è®¾è®¡æ¯æ ¸ç»“æ„50æšï¼Œä¸”å‡å…·å¤‡æœ‰æœºåˆæˆå®éªŒæ–¹æ³•å®ç°çš„å¯èƒ½æ€§
 ; ;
 4. åˆ†å­å¯¹æ¥æ–¹æ³•åˆ¤å®šæœ€ä½³åŒ–å­¦æ¯æ ¸ç»“æ„  åˆ©ç”¨åˆ†å­å¯¹æ¥æ–¹æ³•ï¼Œå°†50æšæ¯æ ¸ç»“æ„åˆ†åˆ«å¯¹æ¥è¿›å…¥çªå˜åè›‹ç™½è´¨XYZç»“æ„ä¸­ï¼Œ åˆ©ç”¨åˆ†å­å¯¹æ¥æƒé‡å€¼, åˆ¤å®š3æšå°åˆ†å­è¯ç‰©åœ¨å¯¹æ¥ç»“æœä¸­å æ®è¾ƒå¥½çš„æƒé‡ä½ç½®ã€‚å¥¥å¤æ–¯æ±€Â·Khanå°†æ­¤ç»“æœå‘ˆç°ç»™ï¼Œç¦æ–¯å¦å›½ç«‹å¤§å­¦åŒ–å­¦é™¢ç ”ç©¶å›¢é˜Ÿï¼Œç»è¿‡å†ä¸‰æ–Ÿé…Œåˆ†æï¼Œå†³å®šé‡‡ç”¨C2ã€C1å’ŒC3è¿›è¡ŒåæœŸåŒ–å­¦ç»“æ„æ”¹é€ 
      5.</div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="/2019/06/deep-work/">
                <h3 class="media-heading">Deep Work</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Jun 6, 2019
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather">æœ€è¿‘åœ¨è¯» å¡å°”.çº½æ³¢ç‰¹çš„ æ·±åº¦å·¥ä½œ, ä»¥ä¸‹æ˜¯é˜…è¯»çš„ä¸€ç‚¹æ„Ÿæ‚Ÿå’Œç¬”è®°.
 æ·±åº¦å·¥ä½œæ˜¯ 21 ä¸–çºªçš„è¶…çº§åŠ›é‡
 ä½œè€…ä»‹ç»: å¡å°”Â·çº½æ³¢ç‰¹ï¼Œç•…é”€ä¹¦ä½œå®¶ï¼Œäººæ°”åšä¸»ï¼Œåˆ›åŠäº†åœ¨ç¾å›½å¾ˆå—æ¬¢è¿çš„åšå®¢â€œå­¦ä¹ é»‘å®¢â€ï¼Œç ´è§£å·¥ä½œå’Œå­¦ä¹ é¢†åŸŸçš„æˆåŠŸæ¨¡å¼ã€‚
 å…³äºæœ¬ä¹¦ è¿™æ˜¯ä¸€æœ¬è‡ªæˆ‘ç®¡ç†ç±»ä¹¦ç±ï¼Œè®²è¿°äº†åœ¨ç¢ç‰‡åŒ–æ—¶ä»£ï¼Œå¦‚ä½•è®­ç»ƒå¤§è„‘æ’é™¤å¹²æ‰°ï¼Œæé«˜å¤§è„‘çš„æ·±åº¦æ€ç»´èƒ½åŠ›ï¼Œåˆ›é€ æ›´å¤šä»·å€¼ã€‚è¿™æœ¬ä¹¦çš„è‹±æ–‡åŸç‰ˆ2016å¹´åœ¨ç¾å›½å‡ºç‰ˆï¼Œä¸€å‘è¡Œå°±å æ®äº†äºšé©¬é€Šç¾å›½ç½‘ç«™çš„èŒåœºåŠ±å¿—ä¹¦æ¦œé¦–ã€‚
å‡ ä¸ªé—®é¢˜:
åœ¨èŒåœºä¸­ï¼Œä¸ºä»€ä¹ˆæœ‰äº›äººè¶Šå¿™ç¢Œè¶Šæ— æ³•äº§å‡ºæœ‰ä»·å€¼çš„æˆæœï¼Ÿ åˆè¯¥å¦‚ä½•é€šè¿‡è®­ç»ƒå¤§è„‘æ’é™¤å¹²æ‰°ï¼Œæé«˜æ·±åº¦æ€ç»´èƒ½åŠ›ï¼Œåˆ›é€ æ›´å¤šä»·å€¼ï¼Ÿ  è¿™æœ¬ä¹¦ä»2ä¸ªæ–¹é¢è§£é‡Šäº†å¤§éƒ¨åˆ†äººæ— æ³•è¿›è¡Œæ·±åº¦å·¥ä½œçš„åŸå› ï¼Œå¹¶ç»™å‡ºäº†åŸ¹å…»æ·±åº¦å·¥ä½œçš„4ä¸ªæ­¥éª¤ï¼Œåˆ†åˆ«æ˜¯:
é€‰æ‹©é€‚åˆè‡ªå·±çš„æ·±åº¦å·¥ä½œæ¨¡å¼ å°†å·¥ä½œå†…åŒ–æˆä¹ æƒ¯ åƒç»å•†ä¸€æ ·å»æ‰§è¡Œ é€‚å½“å‡å°‘æ•´ä½“å·¥ä½œæ—¶é—´éµå¾ªè¿™å‡ ä¸ªæ­¥éª¤è¿›è¡Œåˆ»æ„ç»ƒä¹    æ¦‚å¿µä»‹ç» æ·±åº¦å·¥ä½œ(Deep Work): åœ¨æ²¡æœ‰å¹²æ‰°çš„æƒ…å†µä¸‹ä¸“æ³¨çš„è¿›è¡ŒèŒä¸šæ´»åŠ¨, ä½¿ä¸ªäººçš„è®¤çŸ¥èƒ½åŠ›è¾¾åˆ°æé™, è¿™ç§åŠªåŠ›èƒ½å¤Ÿåˆ›é€ æ–°ä»·å€¼, æå‡æŠ€èƒ½, è€Œä¸”éš¾ä»¥å¤åˆ¶.
è‚¤æµ…å·¥ä½œ(shallow work): å¯¹è®¤çŸ¥è¦æ±‚ä¸é«˜çš„ä»»åŠ¡, åœ¨æ”¶åˆ°å¹²æ‰°çš„æƒ…å†µä¸‹ä¹Ÿèƒ½è¿›è¡Œ, æ­¤ç±»å·¥ä½œåˆ›é€ çš„ä»·å€¼ä¸é«˜, ä¸”å®¹æ˜“å¤åˆ¶.
 æ·±åº¦å·¥ä½œçš„é‡è¦æ€§ ä½œè€…å‰é¢èŠ±äº†å¤§é‡ç¯‡å¹…è¯´æ˜æ·±åº¦å·¥ä½œçš„é‡è¦æ€§, å…¶ä¸­å…³äºæ³¨æ„çš„è®ºæ–­æ·±ä»¥ä¸ºç„¶.
å¿«é€Ÿå­¦ä¹ å¤æ‚çš„æŠ€èƒ½, è¿™èƒ½ä¸ºæˆ‘ä»¬å¸¦æ¥ä»·å€¼, è¿™ä»¶äº‹éœ€è¦æ·±åº¦å·¥ä½œ.
ä½†æ˜¯ç°å®ç”Ÿæ´»ä¸­, ç½‘ç»œå·¥å…·ä½¿æˆ‘ä»¬åˆ†å¿ƒ, å¯¼è‡´ä¸“æ³¨èƒ½åŠ›çš„ä¸‹é™. ä¸åˆ†å¿ƒæ˜¯å¾ˆéš¾çš„, æˆ‘ä»¬éƒ½æœ‰ä¸€ç§å†²åŠ¨å°±æ˜¯, æŠŠè‡ªå·±çš„æ³¨æ„åŠ›è½¬ç§»åˆ°è‚¤æµ…çš„äº‹ç‰©ä¸Š.
æ¯”å¦‚, å·¥ä½œç´¯äº†æˆ–è€…é‡åˆ°éš¾é¢˜äº†å°±è¦åˆ·åˆ·ç¤¾äº¤ç½‘ç»œ, ä½†æ˜¯è¿™ç§è¡Œä¸ºå…¶å®è¿˜æ˜¯åœ¨æ¶ˆè€—ç€ä½ çš„æ³¨æ„åŠ›å’Œèƒ½é‡. ä½ çš„æ„å¿—åŠ›æ˜¯æœ‰é™çš„, å®ƒåœ¨ä½¿ç”¨çš„è¿‡ç¨‹ä¸­æ˜¯ä¸æ–­çš„è¢«æ¶ˆè€—çš„. è€Œè¿›å…¥æ·±åº¦å·¥ä½œçŠ¶æ€æ˜¯éœ€è¦æ„å¿—åŠ›èƒ½é‡, å¦‚ä½•ä½¿å¾—è¿™ä¸ªè½¬åŒ–è¿‡ç¨‹å˜å¾—å®¹æ˜“ä½¿æˆ‘ä»¬åº”è¯¥æŒæ¡çš„æŠ€å·§. ç®€è€Œè¨€ä¹‹, æˆ‘ä»¬å¢åŠ æ·±åº¦å·¥ä½œçš„è¯„ç‡, è€Œå‡å°è½¬ç§»åˆ°è‚¤æµ…å·¥ä½œçš„å†²åŠ¨å’Œé¢‘ç‡. æˆ‘ä»¬éƒ½æœ‰è¿‡ç±»ä¼¼çš„ç»å†, æ‰“å¼€ word æ–‡æ¡£, å‡†å¤‡å†™è®ºæ–‡å’ŒæŠ¥å‘Š, ä¸€ç¬é—´è„‘è¢‹ç©ºç™½, è¿™æ—¶å€™å¹³æ—¶ä¸ç›¸å¹²çš„äº‹æƒ…, çªç„¶éƒ½å˜å¾—å¯çˆ±èµ·æ¥, æ¯”å¦‚: æ´—å®Œ, æ‰«åœ°, æ´—è¡£æœ, æ”¶æ‹¾æˆ¿é—´, ä¸‹æ¥¼ä¹°ğŸ’Š, æ€»ä¹‹åªè¦æ˜¯ä¸æ˜¯å†™è®ºæ–‡, ä»€ä¹ˆå…¶ä»–èƒ½æ‹–å»¶è¿™ä»¶äº‹æƒ…çš„äº‹, æˆ‘ä»¬éƒ½æ„¿æ„å¹².</div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="/2019/05/the-hello-world-of-neural-network/">
                <h3 class="media-heading">The Hello World Of Neural Network</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  May 5, 2019
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather">ç®€å•æ¥è¯´ç¥ç»ç½‘ç»œå’Œæˆ‘ä»¬ä¸€èˆ¬çš„ç¼–ç¨‹åŒºåˆ«åœ¨äº: ä¸€ä¸ªæ˜¯è¾“å…¥æ•°æ®å’Œå‡½æ•°è§„åˆ™, ç„¶åå¾—åˆ°ç»“æœ. è€Œç¥ç»ç½‘ç»œæ˜¯è¾“å…¥æ•°æ®å’Œç­”æ¡ˆ, é€šè¿‡è¿­ä»£å­¦ä¹ , ç¥ç»ç½‘ç»œèƒ½å­¦ä¹ å‡ºå‡½æ•°è§„åˆ™.å¦‚ä¸‹å›¾:
ä¸¾ä¸ªç®€å•çš„ä¾‹å­, è¿™é‡Œæœ‰ä¸¤ç»„æ•°æ®:
X: -1, 0, 1, 2, 3, 4 Y: -3, -1, 1, 3, 5, 7
ä½ å¯ä»¥æŠŠ x çœ‹åšæ˜¯æ•°æ®, y çœ‹åšæ˜¯ç­”æ¡ˆ, ç°åœ¨ä½ è¦åšçš„æ˜¯æ‰¾åˆ°å…¶ä¸­çš„å‡½æ•°å…³ç³», è¿™ä¸ªå…³ç³»èƒ½å¤Ÿå¸®åŠ©æˆ‘ä»¬, ç”¨ x å»é¢„æµ‹ Y çš„å€¼(å‡è®¾ä½ æ²¡æœ‰å­¦è¿‡è§£æ–¹ç¨‹ç»„).
æœ€å¸¸ç”¨çš„æ–¹æ³•å°±æ˜¯å½’çº³æ³•, é¦–å…ˆä½ æ ¹æ®ç¬¬ä¸€å¯¹æ•°æ®çŒœä¸€ä¸ªå¯¹åº”å…³ç³»è§„åˆ™, æ‹¿ç€è¿™ä¸ªè§„åˆ™è®¡ç®—ç­”æ¡ˆå€¼, è¯„ä¼°è®¡ç®—çš„ç­”æ¡ˆå’ŒçœŸå®ç­”æ¡ˆå·®å¤šè¿œ, ç„¶ååœ¨è°ƒæ•´ä½ çš„è§„åˆ™, ç»§ç»­è¯„ä¼°, ç›´åˆ°ä½ çš„è§„åˆ™èƒ½å¤Ÿæ‹Ÿåˆæ‰€æœ‰çš„æ•°æ®.
è¿™å°±æ˜¯ç¥ç»ç½‘ç»œçš„é€»è¾‘è¿‡ç¨‹.
æˆ‘ä»¬æ¥çœ‹ä¸€ä¸ªç®€å•çš„ç¥ç»ç½‘ç»œçš„ä¾‹å­
1.Import: åŠ è½½æ‰€éœ€æ¨¡å— import tensorflow as tf ## /Users/zero/anaconda3/envs/tfdeeplearning/lib/python3.5/importlib/_bootstrap.py:222: RuntimeWarning: compiletime version 3.6 of module &#39;tensorflow.python.framework.fast_tensor_util&#39; does not match runtime version 3.5 ## return f(*args, **kwds) import numpy as np from tensorflow import keras  2.</div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="/2019/04/analyze-and-visualize-your-iphone-s-health-app-data-in-r/">
                <h3 class="media-heading">ä½¿ç”¨ R åˆ†æå¯è§†åŒ–ä½ çš„ iPhone å¥åº· APP æ•°æ®</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Apr 4, 2019
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather">ä»‹ç» iPhone çš„ health APP å­˜å‚¨ç€æˆ‘ä»¬çš„ç§äººå¥åº·æ•°æ®, è¿™é‡Œæœ‰ä¸€ç¯‡å¸–å­æ˜¯ç”¨ Python åˆ†æ health APP çš„æ•°æ®Apple Health Data How to Export Analyze Visualize Guide - ryanpraski.com , è€Œæˆ‘æ›´å–œæ¬¢ R çš„ç‰ˆæœ¬.
è®©æˆ‘ä»¬èµ¶ç´§å¼€å§‹å§!!
 é¦–å…ˆè·å–æ•°æ®å¹¶è¯»å– ä»ä½ çš„ health APP åº”ç”¨ä¸­å¯¼å‡ºæ•°æ® åœ¨ R ä¸­è¯»å–æ•°æ®  åŠ è½½åŒ…å¹¶è¯»å…¥æ•°æ® library(XML) library(tidyverse) library(lubridate) library(scales) library(here) library(ggthemes) xml &lt;- xmlParse(here(&quot;data/apple_health_export/export.xml&quot;)) summary(xml) ## $nameCounts ## ## Record ExportDate HealthData Me Workout ## 90037 1 1 1 1 ## ## $numNodes ## [1] 90041 Record æ˜¯æˆ‘çš„ä¸»è¦æ•°æ®, æœ‰ 90,037 æ¡</div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="/2019/03/tao_wang_bei_shang_guang/">
                <h3 class="media-heading">é€ƒå¾€åŒ—ä¸Šå¹¿</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Mar 3, 2019
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather">å¥‹æ–—ï¼Œåº”è¯¥æ˜¯ä¸€ç§ç”Ÿæ´»æ–¹å¼, ç™»å±±å¹¶ä¸å…¨æ˜¯ä¸ºäº†ç™»é¡¶çš„é‚£ä¸€åˆ»
 åªæœ‰ä¸åœå¥‹æ–—çš„äººç”Ÿï¼Œæ‰æœ‰æ„ä¹‰ï¼Œæ‰æ´»çš„æœ‰åŠ²å„¿ï¼Œæ··åƒç­‰æ­»çš„æ—¥å­å…¶å®æ˜¯å¾ˆè‰°éš¾çš„ï¼Œæ— èŠæ­»äº†ï¼Œä¸€å¤´æ‰è¿›äººå †å„¿é‡Œä¸åœçš„æ‰‘æ£±æ‰æœ‰æ„æ€å•Šï¼
é€ƒå¾€åŒ—äº¬ã€ä¸Šæµ·ã€å¹¿å·ã€æ·±åœ³
åœ¨å®¶ä¹¡ï¼Œåœ¨é‚£ä¸ªä½ ç”Ÿé•¿äº†äºŒä¸‰åå¹´çš„å°åŸé•‡ï¼Œå¦‚æœä½ æ‹’ç»è¿‡ä¸Šå’Œæ‰€æœ‰äººä¸€æ ·å¾ªè§„è¹ˆçŸ©çš„ç”Ÿæ´»ï¼Œæ‹’ç»ç»“å©šç”Ÿå­ï¼Œæ‹’ç»ç¨³å®šå·¥ä½œï¼Œæ‹’ç»æ”¾å¼ƒå½±å“ä½ èµšé’±å‡èŒçš„çˆ±å¥½ï¼Œé‚£ä¹ˆä½ å°±å˜æˆäº†ä¸€å¤´å’Œå‘¨å›´çš„ä¸€åˆ‡æ ¼æ ¼ä¸å…¥çš„æ€ªç‰©ã€‚
äºæ˜¯å¤§å®¶è¦æŒ½æ•‘ä½ ï¼Œè¦æ•™è‚²ä½ ï¼Œè¦è®©ä½ å­¦ä¼šè®¤å‘½ï¼Œåœæ­¢æ¯«æ— æ„ä¹‰çš„æŠ˜è…¾ï¼Œè¿™æ ·ä½ æ‰èƒ½å’Œå¤§å®¶ä¸€æ ·â€œè¸è¸å®å®è¿‡æ—¥å­â€ã€‚
è®¸å¤šäººå› æ­¤é€‰æ‹©äº†å¯¹æŠ—ï¼Œåœ¨æ„¤æ€’ä¸­æ…¢æ…¢å˜å¾—ç»æœ›ï¼Œåœ¨ç»æœ›ä¸­å˜å¾—éº»æœ¨ï¼Œåœ¨éº»æœ¨ä¸­é€ƒå¾€è™šæ— </div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="/2018/10/pubchunks_extract_parts_of_scholarly_xml_articles/">
                <h3 class="media-heading">pubchunksï¼šæå–å­¦æœ¯æ–‡ç« çš„æŸéƒ¨åˆ†</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Oct 10, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather">èƒŒæ™¯ pubchunks çš„ç›®çš„æ˜¯ä»XMLæ ¼å¼çš„å­¦æœ¯æ–‡ç« ä¸­è·å–éƒ¨åˆ†æ•°æ®ã€‚æˆ‘ä»¬ä¸éœ€è¦äº†è§£XMLåŠå…¶æ‰€æœ‰çš„æ ¼å¼åŸç†ã€‚åªéœ€çŸ¥é“æˆ‘ä»¬çš„æ–‡ä»¶æˆ–XMLå­—ç¬¦ä¸²åœ¨å“ªé‡Œä»¥åŠæˆ‘ä»¬æƒ³è¦æ¯ç¯‡æ–‡ç« çš„å“ªäº›éƒ¨åˆ†ã€‚ç„¶åç”¨æˆ·å¯ä»¥ç»„åˆè¿™äº›éƒ¨åˆ†å¹¶åšæˆ‘ä»¬å¸Œæœ›ä¸‹æ¸¸çš„ä»»ä½•äº‹æƒ…; ä¾‹å¦‚,åˆ†ææ–‡æœ¬ç»“æ„
 pubchunksä¸­çš„å‡½æ•° ä¸¤ä¸ªä¸»è¦åŠŸèƒ½æ˜¯ï¼š
pub_chunks()ï¼šè·å–XMLéƒ¨åˆ† pub_tabularize()ï¼šå¼ºåˆ¶è¾“å‡ºpub_chunks()åˆ°data.frameä¸­ pub_guess_publisher()ï¼šä»XMLæ–‡ä»¶æˆ–å­—ç¬¦ä¸²çŒœæµ‹å‘å¸ƒè€… pub_sections()ï¼šéƒ¨åˆ†pubchunksçŸ¥é“å¦‚ä½•å¤„ç† pub_providers()ï¼šæä¾›è€…ï¼ˆå³å‘å¸ƒè€…ï¼‰pubchunksçŸ¥é“å¦‚ä½•æ˜ç¡®å¤„ç†   æ”¯æŒçš„å‡ºç‰ˆå•† elife plos elsevier hindawi pensoft peerj copernicus frontiers f1000research   æ”¯æŒæå–çš„éƒ¨åˆ†æœ‰: Front - å‘å¸ƒè€…ï¼ŒæœŸåˆŠå’Œæ–‡ç« å…ƒæ•°æ®å…ƒç´  Body - æ–‡ç« çš„æ­£æ–‡ Back - æ–‡ç« çš„èƒŒé¢ï¼Œè‡´è°¢ï¼Œä½œè€…è´¡çŒ®ï¼Œå‚è€ƒæ–‡çŒ® Title - æ–‡ç« æ ‡é¢˜ Doi - æ–‡ç« doi Categories - å‘å¸ƒå•†çš„ç±»åˆ«ï¼Œå¦‚æœæœ‰çš„è¯ Author - ä½œè€… Aff - éš¶å±å…³ç³»ï¼ˆåŒ…æ‹¬ä½œè€…å§“åï¼‰ Keyword - å…³é”®å­— Abstract - æ–‡ç« æ‘˜è¦ Executive_summary - æ–‡ç« æ‰§è¡Œæ‘˜è¦ Refs - å‚è€ƒæ–‡çŒ® Refs_dois - å‚è€ƒdois - å¦‚æœæœ‰çš„è¯ Publisher - å‘å¸ƒè€…åç§° Journal_meta - æœŸåˆŠå…ƒæ•°æ® Article_meta - æ–‡ç« å…ƒæ•°æ® Acknowledgments - è‡´è°¢ Permissions - æ–‡ç« æƒé™ History - æ—¥æœŸï¼Œæ”¶åˆ°ï¼Œå‡ºç‰ˆï¼Œæ¥å—ç­‰   å®‰è£… #install.</div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="/2018/09/nlp-part04-word-count-use-r/">
                <h3 class="media-heading">NLP part04 word count use R</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Sep 9, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather">What Is Clean Data Each variable is a column Each observation is a row Each type of observational unit is a table   A table with one-token-per-row.  A token is a meaningful unit of text, such as a word, that we are interested in using for analysis, and tokenization is the process of splitting text into tokens.
 text &lt;- c( &quot;Because I could not stop for Death -&quot;, &quot;He kindly stopped for me -&quot;, &quot;The Carriage held but just Ourselves -&quot;, &quot;and Immortality&quot; ) text ## [1] &quot;Because I could not stop for Death -&quot; ## [2] &quot;He kindly stopped for me -&quot; ## [3] &quot;The Carriage held but just Ourselves -&quot; ## [4] &quot;and Immortality&quot; library(dplyr) text_df &lt;- data_frame(line = 1:4, text = text) text_df ## # A tibble: 4 x 2 ## line text ## &lt;int&gt; &lt;chr&gt; ## 1 1 Because I could not stop for Death - ## 2 2 He kindly stopped for me - ## 3 3 The Carriage held but just Ourselves - ## 4 4 and Immortality library(tidytext) text_df %&gt;% unnest_tokens(word, text) ## # A tibble: 20 x 2 ## line word ## &lt;int&gt; &lt;chr&gt; ## 1 1 because ## 2 1 i ## 3 1 could ## 4 1 not ## 5 1 stop ## 6 1 for ## 7 1 death ## 8 2 he ## 9 2 kindly ## 10 2 stopped ## 11 2 for ## 12 2 me ## 13 3 the ## 14 3 carriage ## 15 3 held ## 16 3 but ## 17 3 just ## 18 3 ourselves ## 19 4 and ## 20 4 immortality For tidy text mining, the token that is stored in each row is most often a single word, but can also be an n-gram, sentence, or paragraph</div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="/2018/09/nlp-part-03-sentiment-analysis/">
                <h3 class="media-heading">NLP part 03 sentiment analysis</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Sep 9, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather">Case Study: Sentiment Analysis Data Prep import pandas as pd import numpy as np # Read in the data df = pd.read_csv(&#39;/Users/zero/Desktop/NLP/raw-data/Amazon_Unlocked_Mobile.csv&#39;) # å¯¹æ•°æ®è¿›è¡Œé‡‡æ ·ä»¥åŠ å¿«è®¡ç®—é€Ÿåº¦ # Comment out this line to match with lecture df = df.sample(frac=0.1, random_state=10) df.head()   .dataframe thead tr:only-child th { text-align: right; } .dataframe thead th { text-align: left; } .dataframe tbody tr th { vertical-align: top; }   &lt;tr style=&quot;text-align: right;&quot;&gt; &lt;th&gt;&lt;/th&gt; &lt;th&gt;Product Name&lt;/th&gt; &lt;th&gt;Brand Name&lt;/th&gt; &lt;th&gt;Price&lt;/th&gt; &lt;th&gt;Rating&lt;/th&gt; &lt;th&gt;Reviews&lt;/th&gt; &lt;th&gt;Review Votes&lt;/th&gt; &lt;/tr&gt;   &lt;tr&gt; &lt;th&gt;394349&lt;/th&gt; &lt;td&gt;Sony XPERIA Z2 D6503 FACTORY UNLOCKED Internat.</div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="/2018/09/nlp-part-03-sentiment-analysis/">
                <h3 class="media-heading">NLP part 03 sentiment analysis</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Sep 9, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
      </div>
    </div>
    <div class="modal-footer">
      <p class="results-count text-medium"
         data-message-zero="no post found"
         data-message-one="1 post found"
         data-message-other="{n} posts found">
         56 posts found
      </p>
    </div>
  </div>
</div>
    
  
    
    <div id="cover" style="background-image:url('https://i.loli.net/2018/02/26/5a9416186c149.jpg');"></div>
  


    
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.1.3/jquery.min.js" integrity="sha256-IFHWFEbU2/+wNycDECKgjIRSirRNIDp2acEB5fvdVRU=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.8.0/highlight.min.js" integrity="sha256-+mpyNVJsNt4rVXCw0F+pAOiB3YxmHgrbJsx4ecPuUaI=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.4/jquery.fancybox.min.js" integrity="sha256-vMxgR/7FtLovVA+IPrR7+xTgIgARH7y9VZQnmmi0HDI=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.4/helpers/jquery.fancybox-thumbs.min.js" integrity="sha256-N0qFUh7/9vLvia87dDndewmsgsyYoNkdA212tPc+2NI=" crossorigin="anonymous"></script>


<script src="/js/script-kr8dyqj6rb6ortyib7whfo9x9p6td6zo8t1v4fdz4ecx5kwybsdlmk1slygn.min.js"></script>


<script>
$(document).ready(function() {
  hljs.configure({ classPrefix: '', useBR: false });
  $('pre.code-highlight, pre > code').each(function(i, block) {
    if (!$(this).hasClass('codeblock')) {
      $(this).addClass('codeblock');
    }
    hljs.highlightBlock(block);
  });
});
</script>

  
    
      <script>
        var disqus_config = function () {
          this.page.url = '\/2018\/04\/how-to-get-genomewide-position-specific-scores\/';
          
            this.page.identifier = '\/2018\/04\/how-to-get-genomewide-position-specific-scores\/'
          
        };
        (function() {
          
          
          if (window.location.hostname == "localhost") {
            return;
          }
          var d = document, s = d.createElement('script');
          var disqus_shortname = 'jixing';
          s.src = '//' + disqus_shortname + '.disqus.com/embed.js';

          s.setAttribute('data-timestamp', +new Date());
          (d.head || d.body).appendChild(s);
        })();
      </script>
    
  



    
  </body>
</html>

